
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<link href="../assets/images/favicon.png" rel="icon"/>
<meta content="mkdocs-1.6.1, mkdocs-material-9.6.22" name="generator"/>
<title>Class: CellTypeTaxon - BICAN Knowledge Graph Models Documentation</title>
<link href="../assets/stylesheets/main.84d31ad4.min.css" rel="stylesheet"/>
<link href="../assets/stylesheets/palette.06af60db.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
</head>
<body data-md-color-accent="indigo" data-md-color-primary="white" data-md-color-scheme="default" dir="ltr">
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#class-celltypetaxon">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="BICAN Knowledge Graph Models Documentation" class="md-header__button md-logo" data-md-component="logo" href=".." title="BICAN Knowledge Graph Models Documentation">
<img alt="logo" src="../images/bican_logo.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            BICAN Knowledge Graph Models Documentation
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Class: CellTypeTaxon
            
          </span>
</div>
</div>
</div>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="" tabindex="0">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/brain-bican/models" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="BICAN Knowledge Graph Models Documentation" class="md-nav__button md-logo" data-md-component="logo" href=".." title="BICAN Knowledge Graph Models Documentation">
<img alt="logo" src="../images/bican_logo.png"/>
</a>
    BICAN Knowledge Graph Models Documentation
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/brain-bican/models" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"></path></svg>
</div>
<div class="md-source__repository">
    GitHub
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="..">
<span class="md-ellipsis">
    Overview
    
  </span>
</a>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" id="__nav_2" type="checkbox"/>
<label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
<span class="md-ellipsis">
    Models
    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_2_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_2">
<span class="md-nav__icon md-icon"></span>
            Models
          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../index_library_generation/">
<span class="md-ellipsis">
    Library Generation
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../index_anatomical_structure/">
<span class="md-ellipsis">
    Anatomical Structure
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../index_genome_annotation/">
<span class="md-ellipsis">
    Genome Annotation
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../index_assertion_evidence/">
<span class="md-ellipsis">
    Assertion Evidence
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../index_bke_taxonomy/">
<span class="md-ellipsis">
    BKE Taxonomy
    
  </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item md-nav__item--nested">
<input class="md-nav__toggle md-toggle" id="__nav_3" type="checkbox"/>
<label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
<span class="md-ellipsis">
    Visualizations
    
  </span>
<span class="md-nav__icon md-icon"></span>
</label>
<nav aria-expanded="false" aria-labelledby="__nav_3_label" class="md-nav" data-md-level="1">
<label class="md-nav__title" for="__nav_3">
<span class="md-nav__icon md-icon"></span>
            Visualizations
          </label>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="../library_generation/">
<span class="md-ellipsis">
    Library Generation
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../anatomical_structure/">
<span class="md-ellipsis">
    Anatomical Structure
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../genome_annotation/">
<span class="md-ellipsis">
    Genome Annotation
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../assertion_evidence/">
<span class="md-ellipsis">
    Assertion Evidence
    
  </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../bke_taxonomy/">
<span class="md-ellipsis">
    BKE Taxonomy
    
  </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#inheritance">
<span class="md-ellipsis">
      Inheritance
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#slots">
<span class="md-ellipsis">
      Slots
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#usages">
<span class="md-ellipsis">
      Usages
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#identifier-and-mapping-information">
<span class="md-ellipsis">
      Identifier and Mapping Information
    </span>
</a>
<nav aria-label="Identifier and Mapping Information" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#schema-source">
<span class="md-ellipsis">
      Schema Source
    </span>
</a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#mappings">
<span class="md-ellipsis">
      Mappings
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#linkml-source">
<span class="md-ellipsis">
      LinkML Source
    </span>
</a>
<nav aria-label="LinkML Source" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#direct">
<span class="md-ellipsis">
      Direct
    </span>
</a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#induced">
<span class="md-ellipsis">
      Induced
    </span>
</a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="class-celltypetaxon">Class: CellTypeTaxon</h1>
<p><em>A cell type taxon is a node in a taxonomy. Taxons represents a unit of cell type classification and is defined by a set of clusters. Taxons may be organized into a heirarchy and levels. Taxons at higher levels represents broader cell type classes and partitioned into more specific types at lower levels. Additionally, taxons may be organized into neighborhoods or related groups.</em></p>
<p>URI: <a href="https://identifiers.org/brain-bican/vocab/CellTypeTaxon">bican:CellTypeTaxon</a></p>
<div class="mermaid"> classDiagram
    class CellTypeTaxon
    click CellTypeTaxon href "../CellTypeTaxon"
      ProvEntity &lt;|-- CellTypeTaxon
        click ProvEntity href "../ProvEntity"
      NamedThing &lt;|-- CellTypeTaxon
        click NamedThing href "../NamedThing"

      CellTypeTaxon : accession_id

      CellTypeTaxon : category

      CellTypeTaxon : contains_cluster




    CellTypeTaxon --&gt; "*" Cluster : contains_cluster
    click Cluster href "../Cluster"


      CellTypeTaxon : curated_markers_to_mouse




    CellTypeTaxon --&gt; "*" GeneAnnotation : curated_markers_to_mouse
    click GeneAnnotation href "../GeneAnnotation"


      CellTypeTaxon : curated_markers_to_primates




    CellTypeTaxon --&gt; "*" GeneAnnotation : curated_markers_to_primates
    click GeneAnnotation href "../GeneAnnotation"


      CellTypeTaxon : deprecated

      CellTypeTaxon : description

      CellTypeTaxon : equivalent_identifiers

      CellTypeTaxon : full_name

      CellTypeTaxon : has_attribute




    CellTypeTaxon --&gt; "*" Attribute : has_attribute
    click Attribute href "../Attribute"


      CellTypeTaxon : has_abbreviation




    CellTypeTaxon --&gt; "*" Abbreviation : has_abbreviation
    click Abbreviation href "../Abbreviation"


      CellTypeTaxon : has_parent




    CellTypeTaxon --&gt; "0..1" CellTypeTaxon : has_parent
    click CellTypeTaxon href "../CellTypeTaxon"


      CellTypeTaxon : id

      CellTypeTaxon : information_content

      CellTypeTaxon : iri

      CellTypeTaxon : name

      CellTypeTaxon : named_thing_category

      CellTypeTaxon : number_of_cells

      CellTypeTaxon : order

      CellTypeTaxon : part_of_taxonomy




    CellTypeTaxon --&gt; "0..1" CellTypeTaxonomy : part_of_taxonomy
    click CellTypeTaxonomy href "../CellTypeTaxonomy"


      CellTypeTaxon : provided_by

      CellTypeTaxon : spatial_proportions_human




    CellTypeTaxon --&gt; "0..1" SpatialProportions : spatial_proportions_human
    click SpatialProportions href "../SpatialProportions"


      CellTypeTaxon : spatial_proportions_macaque




    CellTypeTaxon --&gt; "0..1" SpatialProportions : spatial_proportions_macaque
    click SpatialProportions href "../SpatialProportions"


      CellTypeTaxon : spatial_proportions_marmoset




    CellTypeTaxon --&gt; "0..1" SpatialProportions : spatial_proportions_marmoset
    click SpatialProportions href "../SpatialProportions"


      CellTypeTaxon : spatial_regional_proportions




    CellTypeTaxon --&gt; "0..1" SpatialProportions : spatial_regional_proportions
    click SpatialProportions href "../SpatialProportions"


      CellTypeTaxon : synonym

      CellTypeTaxon : type

      CellTypeTaxon : was_derived_from




    CellTypeTaxon --&gt; "0..1" ProvEntity : was_derived_from
    click ProvEntity href "../ProvEntity"


      CellTypeTaxon : was_generated_by




    CellTypeTaxon --&gt; "0..1" ProvActivity : was_generated_by
    click ProvActivity href "../ProvActivity"


      CellTypeTaxon : xref


</div>
<h2 id="inheritance">Inheritance</h2>
<ul>
<li><a href="../Entity/">Entity</a><ul>
<li><a href="../NamedThing/">NamedThing</a><ul>
<li><strong>CellTypeTaxon</strong> [ <a href="../ProvEntity/">ProvEntity</a>]</li>
</ul>
</li>
</ul>
</li>
</ul>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Name</th>
<th>Cardinality and Range</th>
<th>Description</th>
<th>Inheritance</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../part_of_taxonomy/">part_of_taxonomy</a></td>
<td>0..1 <br/> <a href="../CellTypeTaxonomy/">CellTypeTaxonomy</a></td>
<td>The cell types taxonomy for which the cell type taxon is part of</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../has_parent/">has_parent</a></td>
<td>0..1 <br/> <a href="./">CellTypeTaxon</a></td>
<td>When the taxonomy is a single inheritance tree, the parent represents the sup...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../has_abbreviation/">has_abbreviation</a></td>
<td>* <br/> <a href="../Abbreviation/">Abbreviation</a></td>
<td>One of many abbreviation that are part of the cell type taxon name</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../id/">id</a></td>
<td>1 <br/> <a href="../String/">String</a></td>
<td>( database GUID)</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../accession_id/">accession_id</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>A provider assigned accession identifier for the cell type taxon</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../name/">name</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>Name of the cell type taxon</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../order/">order</a></td>
<td>0..1 <br/> <a href="../Integer/">Integer</a></td>
<td>The priority or display order of the cell type taxon among all other taxons i...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../description/">description</a></td>
<td>0..1 <br/> <a href="../String/">String</a></td>
<td>Description of the cell type taxon (optional)</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../xref/">xref</a></td>
<td>* <br/> <a href="../Uriorcurie/">Uriorcurie</a></td>
<td>A database cross reference or alternative identifier for a NamedThing or edge...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../contains_cluster/">contains_cluster</a></td>
<td>* <br/> <a href="../Cluster/">Cluster</a></td>
<td>One of potentially cell clusters which together defines the cell type taxon</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../spatial_regional_proportions/">spatial_regional_proportions</a></td>
<td>0..1 <br/> <a href="../SpatialProportions/">SpatialProportions</a></td>
<td>Species-independent quantification of regional composition (how much of somet...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../spatial_proportions_marmoset/">spatial_proportions_marmoset</a></td>
<td>0..1 <br/> <a href="../SpatialProportions/">SpatialProportions</a></td>
<td>Quantification of regional composition (how much of something is found in eac...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../spatial_proportions_macaque/">spatial_proportions_macaque</a></td>
<td>0..1 <br/> <a href="../SpatialProportions/">SpatialProportions</a></td>
<td>Quantification of regional composition (how much of something is found in eac...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../spatial_proportions_human/">spatial_proportions_human</a></td>
<td>0..1 <br/> <a href="../SpatialProportions/">SpatialProportions</a></td>
<td>Quantification of regional composition (how much of something is found in eac...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../curated_markers_to_primates/">curated_markers_to_primates</a></td>
<td>* <br/> <a href="../GeneAnnotation/">GeneAnnotation</a></td>
<td>Marker genes in the primate species that are related to the respective cell t...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../curated_markers_to_mouse/">curated_markers_to_mouse</a></td>
<td>* <br/> <a href="../GeneAnnotation/">GeneAnnotation</a></td>
<td>Marker genes in the mouse species that are related to the respective cell typ...</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../number_of_cells/">number_of_cells</a></td>
<td>0..1 <br/> <a href="../Integer/">Integer</a></td>
<td>The aggregated number of cells that defines the cell type taxon</td>
<td>direct</td>
</tr>
<tr>
<td><a href="../was_derived_from/">was_derived_from</a></td>
<td>0..1 <br/> <a href="../ProvEntity/">ProvEntity</a></td>
<td>A derivation is a transformation of an entity into another, an update of an e...</td>
<td><a href="../ProvEntity/">ProvEntity</a></td>
</tr>
<tr>
<td><a href="../was_generated_by/">was_generated_by</a></td>
<td>0..1 <br/> <a href="../ProvActivity/">ProvActivity</a></td>
<td>Generation is the completion of production of a new entity by an activity</td>
<td><a href="../ProvEntity/">ProvEntity</a></td>
</tr>
<tr>
<td><a href="../iri/">iri</a></td>
<td>0..1 <br/> <a href="../IriType/">IriType</a></td>
<td>An IRI for an entity</td>
<td><a href="../NamedThing/">NamedThing</a>, <a href="../Entity/">Entity</a></td>
</tr>
<tr>
<td><a href="../type/">type</a></td>
<td>* <br/> <a href="../String/">String</a></td>
<td></td>
<td><a href="../NamedThing/">NamedThing</a>, <a href="../Entity/">Entity</a></td>
</tr>
<tr>
<td><a href="../has_attribute/">has_attribute</a></td>
<td>* <br/> <a href="../Attribute/">Attribute</a></td>
<td>connects any entity to an attribute</td>
<td><a href="../NamedThing/">NamedThing</a>, <a href="../Entity/">Entity</a></td>
</tr>
<tr>
<td><a href="../deprecated/">deprecated</a></td>
<td>0..1 <br/> <a href="../Boolean/">Boolean</a></td>
<td>A boolean flag indicating that an entity is no longer considered current or v...</td>
<td><a href="../NamedThing/">NamedThing</a>, <a href="../Entity/">Entity</a></td>
</tr>
<tr>
<td><a href="../provided_by/">provided_by</a></td>
<td>* <br/> <a href="../String/">String</a></td>
<td>The value in this node property represents the knowledge provider that create...</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
<tr>
<td><a href="../full_name/">full_name</a></td>
<td>0..1 <br/> <a href="../LabelType/">LabelType</a></td>
<td>a long-form human readable name for a thing</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
<tr>
<td><a href="../synonym/">synonym</a></td>
<td>* <br/> <a href="../LabelType/">LabelType</a></td>
<td>Alternate human-readable names for a thing</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
<tr>
<td><a href="../information_content/">information_content</a></td>
<td>0..1 <br/> <a href="../Float/">Float</a></td>
<td>Information content (IC) value for a term, primarily from Automats</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
<tr>
<td><a href="../equivalent_identifiers/">equivalent_identifiers</a></td>
<td>* <br/> <a href="../Uriorcurie/">Uriorcurie</a></td>
<td>A set of identifiers that are considered equivalent to the primary identifier...</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
<tr>
<td><a href="../named_thing_category/">named_thing_category</a></td>
<td>1..* <br/> <a href="../Uriorcurie/">Uriorcurie</a></td>
<td>Name of the high level ontology class in which this entity is categorized</td>
<td><a href="../NamedThing/">NamedThing</a></td>
</tr>
<tr>
<td><a href="../category/">category</a></td>
<td>1..* <br/> <a href="../Uriorcurie/">Uriorcurie</a></td>
<td>Name of the high level ontology class in which this entity is categorized</td>
<td><a href="../Entity/">Entity</a></td>
</tr>
</tbody>
</table>
<h2 id="usages">Usages</h2>
<table>
<thead>
<tr>
<th>used by</th>
<th>used in</th>
<th>type</th>
<th>used</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../CellTypeSet/">CellTypeSet</a></td>
<td><a href="../contains_taxon/">contains_taxon</a></td>
<td>range</td>
<td><a href="./">CellTypeTaxon</a></td>
</tr>
<tr>
<td><a href="./">CellTypeTaxon</a></td>
<td><a href="../has_parent/">has_parent</a></td>
<td>range</td>
<td><a href="./">CellTypeTaxon</a></td>
</tr>
<tr>
<td><a href="../Abbreviation/">Abbreviation</a></td>
<td><a href="../denotes_cell_type/">denotes_cell_type</a></td>
<td>range</td>
<td><a href="./">CellTypeTaxon</a></td>
</tr>
<tr>
<td><a href="../DisplayColor/">DisplayColor</a></td>
<td><a href="../is_color_for_taxon/">is_color_for_taxon</a></td>
<td>range</td>
<td><a href="./">CellTypeTaxon</a></td>
</tr>
</tbody>
</table>
<h2 id="identifier-and-mapping-information">Identifier and Mapping Information</h2>
<h3 id="schema-source">Schema Source</h3>
<ul>
<li>from schema: https://w3id.org/brain-bican/bke-taxonomy</li>
</ul>
<h2 id="mappings">Mappings</h2>
<table>
<thead>
<tr>
<th>Mapping Type</th>
<th>Mapped Value</th>
</tr>
</thead>
<tbody>
<tr>
<td>self</td>
<td>bican:CellTypeTaxon</td>
</tr>
<tr>
<td>native</td>
<td>bican:CellTypeTaxon</td>
</tr>
</tbody>
</table>
<h2 id="linkml-source">LinkML Source</h2>
<!-- TODO: investigate https://stackoverflow.com/questions/37606292/how-to-create-tabbed-code-blocks-in-mkdocs-or-sphinx -->
<h3 id="direct">Direct</h3>
<details>
<pre><code class="language-yaml">name: CellTypeTaxon
description: A cell type taxon is a node in a taxonomy. Taxons represents a unit of
  cell type classification and is defined by a set of clusters. Taxons may be organized
  into a heirarchy and levels. Taxons at higher levels represents broader cell type
  classes and partitioned into more specific types at lower levels. Additionally,
  taxons may be organized into neighborhoods or related groups.
from_schema: https://w3id.org/brain-bican/bke-taxonomy
is_a: named thing
mixins:
- ProvEntity
slots:
- part_of_taxonomy
- has_parent
- has_abbreviation
- id
- accession_id
- name
- order
- description
- xref
slot_usage:
  part_of_taxonomy:
    name: part_of_taxonomy
    description: The cell types taxonomy for which the cell type taxon is part of.
    range: CellTypeTaxonomy
  has_parent:
    name: has_parent
    description: When the taxonomy is a single inheritance tree, the parent represents
      the superclass for which the cell type taxon is a member of.
    range: CellTypeTaxon
  has_abbreviation:
    name: has_abbreviation
    description: One of many abbreviation that are part of the cell type taxon name.
    range: Abbreviation
    multivalued: true
  id:
    name: id
    description: ( database GUID)
    from_schema: bican_biolink
    range: string
  accession_id:
    name: accession_id
    description: A provider assigned accession identifier for the cell type taxon.
    examples:
    - value: CS20230722_CLAS_11
    range: string
  name:
    name: name
    description: Name of the cell type taxon.
    examples:
    - value: 11 CNU-HYa GABA
    range: string
  order:
    name: order
    local_names:
      allen:
        local_name_source: allen
        local_name_value: order_idx
    description: The priority or display order of the cell type taxon among all other
      taxons in the taxonomy.
    examples:
    - value: '11'
    range: integer
    minimum_value: '0.0'
  description:
    name: description
    description: Description of the cell type taxon (optional).
    from_schema: bican_biolink
    range: string
  xref:
    name: xref
    local_names:
      allen:
        local_name_source: allen
        local_name_value: unique_id
    from_schema: bican_biolink
attributes:
  contains_cluster:
    name: contains_cluster
    description: One of potentially cell clusters which together defines the cell
      type taxon.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    domain_of:
    - CellTypeTaxon
    range: Cluster
    multivalued: true
  spatial_regional_proportions:
    name: spatial_regional_proportions
    description: Species-independent quantification of regional composition (how much
      of something is found in each named brain region),
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    domain_of:
    - CellTypeTaxon
    range: SpatialProportions
  spatial_proportions_marmoset:
    name: spatial_proportions_marmoset
    description: Quantification of regional composition (how much of something is
      found in each named brain region) in Marmoset species.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    domain_of:
    - CellTypeTaxon
    range: SpatialProportions
  spatial_proportions_macaque:
    name: spatial_proportions_macaque
    description: Quantification of regional composition (how much of something is
      found in each named brain region) in Macaque species.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    domain_of:
    - CellTypeTaxon
    range: SpatialProportions
  spatial_proportions_human:
    name: spatial_proportions_human
    description: Quantification of regional composition (how much of something is
      found in each named brain region) in Human species.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    domain_of:
    - CellTypeTaxon
    range: SpatialProportions
  curated_markers_to_primates:
    name: curated_markers_to_primates
    description: Marker genes in the primate species that are related to the respective
      cell type.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    domain_of:
    - CellTypeTaxon
    range: GeneAnnotation
    multivalued: true
  curated_markers_to_mouse:
    name: curated_markers_to_mouse
    description: Marker genes in the mouse species that are related to the respective
      cell type.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    domain_of:
    - CellTypeTaxon
    range: GeneAnnotation
    multivalued: true
  number_of_cells:
    name: number_of_cells
    description: The aggregated number of cells that defines the cell type taxon.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    domain_of:
    - CellTypeTaxon
    range: integer
    minimum_value: '0.0'

</code></pre>
</details>
<h3 id="induced">Induced</h3>
<details>
<pre><code class="language-yaml">name: CellTypeTaxon
description: A cell type taxon is a node in a taxonomy. Taxons represents a unit of
  cell type classification and is defined by a set of clusters. Taxons may be organized
  into a heirarchy and levels. Taxons at higher levels represents broader cell type
  classes and partitioned into more specific types at lower levels. Additionally,
  taxons may be organized into neighborhoods or related groups.
from_schema: https://w3id.org/brain-bican/bke-taxonomy
is_a: named thing
mixins:
- ProvEntity
slot_usage:
  part_of_taxonomy:
    name: part_of_taxonomy
    description: The cell types taxonomy for which the cell type taxon is part of.
    range: CellTypeTaxonomy
  has_parent:
    name: has_parent
    description: When the taxonomy is a single inheritance tree, the parent represents
      the superclass for which the cell type taxon is a member of.
    range: CellTypeTaxon
  has_abbreviation:
    name: has_abbreviation
    description: One of many abbreviation that are part of the cell type taxon name.
    range: Abbreviation
    multivalued: true
  id:
    name: id
    description: ( database GUID)
    from_schema: bican_biolink
    range: string
  accession_id:
    name: accession_id
    description: A provider assigned accession identifier for the cell type taxon.
    examples:
    - value: CS20230722_CLAS_11
    range: string
  name:
    name: name
    description: Name of the cell type taxon.
    examples:
    - value: 11 CNU-HYa GABA
    range: string
  order:
    name: order
    local_names:
      allen:
        local_name_source: allen
        local_name_value: order_idx
    description: The priority or display order of the cell type taxon among all other
      taxons in the taxonomy.
    examples:
    - value: '11'
    range: integer
    minimum_value: '0.0'
  description:
    name: description
    description: Description of the cell type taxon (optional).
    from_schema: bican_biolink
    range: string
  xref:
    name: xref
    local_names:
      allen:
        local_name_source: allen
        local_name_value: unique_id
    from_schema: bican_biolink
attributes:
  contains_cluster:
    name: contains_cluster
    description: One of potentially cell clusters which together defines the cell
      type taxon.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: contains_cluster
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxon
    range: Cluster
    multivalued: true
  spatial_regional_proportions:
    name: spatial_regional_proportions
    description: Species-independent quantification of regional composition (how much
      of something is found in each named brain region),
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: spatial_regional_proportions
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxon
    range: SpatialProportions
  spatial_proportions_marmoset:
    name: spatial_proportions_marmoset
    description: Quantification of regional composition (how much of something is
      found in each named brain region) in Marmoset species.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: spatial_proportions_marmoset
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxon
    range: SpatialProportions
  spatial_proportions_macaque:
    name: spatial_proportions_macaque
    description: Quantification of regional composition (how much of something is
      found in each named brain region) in Macaque species.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: spatial_proportions_macaque
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxon
    range: SpatialProportions
  spatial_proportions_human:
    name: spatial_proportions_human
    description: Quantification of regional composition (how much of something is
      found in each named brain region) in Human species.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: spatial_proportions_human
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxon
    range: SpatialProportions
  curated_markers_to_primates:
    name: curated_markers_to_primates
    description: Marker genes in the primate species that are related to the respective
      cell type.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: curated_markers_to_primates
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxon
    range: GeneAnnotation
    multivalued: true
  curated_markers_to_mouse:
    name: curated_markers_to_mouse
    description: Marker genes in the mouse species that are related to the respective
      cell type.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: curated_markers_to_mouse
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxon
    range: GeneAnnotation
    multivalued: true
  number_of_cells:
    name: number_of_cells
    description: The aggregated number of cells that defines the cell type taxon.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: number_of_cells
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxon
    range: integer
    minimum_value: '0.0'
  part_of_taxonomy:
    name: part_of_taxonomy
    description: The cell types taxonomy for which the cell type taxon is part of.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: part_of_taxonomy
    owner: CellTypeTaxon
    domain_of:
    - CellTypeSet
    - CellTypeTaxon
    range: CellTypeTaxonomy
  has_parent:
    name: has_parent
    description: When the taxonomy is a single inheritance tree, the parent represents
      the superclass for which the cell type taxon is a member of.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: has_parent
    owner: CellTypeTaxon
    domain_of:
    - CellTypeSet
    - CellTypeTaxon
    range: CellTypeTaxon
  has_abbreviation:
    name: has_abbreviation
    description: One of many abbreviation that are part of the cell type taxon name.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: has_abbreviation
    owner: CellTypeTaxon
    domain_of:
    - CellTypeSet
    - CellTypeTaxon
    range: Abbreviation
    multivalued: true
  id:
    name: id
    definition_uri: https://w3id.org/biolink/vocab/id
    description: ( database GUID)
    in_subset:
    - translator_minimal
    from_schema: bican_biolink
    exact_mappings:
    - AGRKB:primaryId
    - gff3:ID
    - gpi:DB_Object_ID
    rank: 1000
    domain: entity
    slot_uri: biolink:id
    identifier: true
    alias: id
    owner: CellTypeTaxon
    domain_of:
    - ontology class
    - entity
    - CellTypeTaxonomy
    - CellTypeSet
    - CellTypeTaxon
    - ClusterSet
    - Cluster
    - Abbreviation
    - ColorPalette
    - DisplayColor
    - attribute
    - named thing
    - taxonomic rank
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: string
    required: true
  accession_id:
    name: accession_id
    description: A provider assigned accession identifier for the cell type taxon.
    examples:
    - value: CS20230722_CLAS_11
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: accession_id
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxonomy
    - CellTypeSet
    - CellTypeTaxon
    - ClusterSet
    - Cluster
    range: string
  name:
    name: name
    definition_uri: https://w3id.org/biolink/vocab/name
    description: Name of the cell type taxon.
    examples:
    - value: 11 CNU-HYa GABA
    in_subset:
    - translator_minimal
    - samples
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    aliases:
    - label
    - display name
    - title
    mappings:
    - rdfs:label
    exact_mappings:
    - gff3:Name
    - gpi:DB_Object_Name
    narrow_mappings:
    - dct:title
    - WIKIDATA_PROPERTY:P1476
    rank: 1000
    domain: entity
    slot_uri: rdfs:label
    alias: name
    owner: CellTypeTaxon
    domain_of:
    - attribute
    - entity
    - macromolecular machine mixin
    - CellTypeTaxonomy
    - CellTypeSet
    - CellTypeTaxon
    - ClusterSet
    - Cluster
    - ColorPalette
    - named thing
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: string
  order:
    name: order
    local_names:
      allen:
        local_name_source: allen
        local_name_value: order_idx
    description: The priority or display order of the cell type taxon among all other
      taxons in the taxonomy.
    examples:
    - value: '11'
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    alias: order
    owner: CellTypeTaxon
    domain_of:
    - CellTypeSet
    - CellTypeTaxon
    range: integer
    minimum_value: '0.0'
  description:
    name: description
    definition_uri: https://w3id.org/biolink/vocab/description
    description: Description of the cell type taxon (optional).
    in_subset:
    - translator_minimal
    from_schema: bican_biolink
    aliases:
    - definition
    mappings:
    - dct:description
    exact_mappings:
    - IAO:0000115
    - skos:definitions
    narrow_mappings:
    - gff3:Description
    rank: 1000
    slot_uri: dct:description
    alias: description
    owner: CellTypeTaxon
    domain_of:
    - entity
    - CellTypeTaxonomy
    - CellTypeSet
    - CellTypeTaxon
    - ClusterSet
    - ColorPalette
    - attribute
    - named thing
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: string
  xref:
    name: xref
    definition_uri: https://w3id.org/biolink/vocab/xref
    local_names:
      allen:
        local_name_source: allen
        local_name_value: unique_id
    description: A database cross reference or alternative identifier for a NamedThing
      or edge between two NamedThings.  This property should point to a database record
      or webpage that supports the existence of the edge, or gives more detail about
      the edge. This property can be used on a node or edge to provide multiple URIs
      or CURIE cross references.
    in_subset:
    - translator_minimal
    from_schema: bican_biolink
    aliases:
    - dbxref
    - Dbxref
    - DbXref
    - record_url
    - source_record_urls
    narrow_mappings:
    - gff3:Dbxref
    - gpi:DB_Xrefs
    rank: 1000
    domain: named thing
    slot_uri: biolink:xref
    alias: xref
    owner: CellTypeTaxon
    domain_of:
    - named thing
    - gene
    - CellTypeTaxonomy
    - CellTypeSet
    - CellTypeTaxon
    - ClusterSet
    - Cluster
    - ObservationMatrix
    - ObservationRow
    - Abbreviation
    - MatrixFile
    - ColorPalette
    - DisplayColor
    - attribute
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - genome
    range: uriorcurie
    multivalued: true
  was_derived_from:
    name: was_derived_from
    description: A derivation is a transformation of an entity into another, an update
      of an entity resulting in a new one, or the construction of a new entity based
      on a pre-existing entity.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    slot_uri: prov:wasDerivedFrom
    alias: was_derived_from
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxonomy
    - ClusterSet
    - ObservationMatrix
    - ObservationRow
    - ProvEntity
    range: ProvEntity
  was_generated_by:
    name: was_generated_by
    description: Generation is the completion of production of a new entity by an
      activity. This entity did not exist before generation and becomes available
      for usage after this generation.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    slot_uri: prov:wasGeneratedBy
    alias: was_generated_by
    owner: CellTypeTaxon
    domain_of:
    - CellTypeTaxonomy
    - ClusterSet
    - ObservationMatrix
    - ProvEntity
    range: ProvActivity
  iri:
    name: iri
    definition_uri: https://w3id.org/biolink/vocab/iri
    description: An IRI for an entity. This is determined by the id using expansion
      rules.
    in_subset:
    - translator_minimal
    - samples
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    exact_mappings:
    - WIKIDATA_PROPERTY:P854
    rank: 1000
    slot_uri: biolink:iri
    alias: iri
    owner: CellTypeTaxon
    domain_of:
    - attribute
    - entity
    - named thing
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: iri type
  type:
    name: type
    definition_uri: https://w3id.org/biolink/vocab/type
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    mappings:
    - rdf:type
    exact_mappings:
    - gff3:type
    - gpi:DB_Object_Type
    rank: 1000
    domain: entity
    slot_uri: rdf:type
    alias: type
    owner: CellTypeTaxon
    domain_of:
    - entity
    - attribute
    - named thing
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: string
    multivalued: true
  has attribute:
    name: has attribute
    definition_uri: https://w3id.org/biolink/vocab/has_attribute
    description: connects any entity to an attribute
    in_subset:
    - samples
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    exact_mappings:
    - SIO:000008
    close_mappings:
    - OBI:0001927
    narrow_mappings:
    - OBAN:association_has_subject_property
    - OBAN:association_has_object_property
    - CPT:has_possibly_included_panel_element
    - DRUGBANK:category
    - EFO:is_executed_in
    - HANCESTRO:0301
    - LOINC:has_action_guidance
    - LOINC:has_adjustment
    - LOINC:has_aggregation_view
    - LOINC:has_approach_guidance
    - LOINC:has_divisor
    - LOINC:has_exam
    - LOINC:has_method
    - LOINC:has_modality_subtype
    - LOINC:has_object_guidance
    - LOINC:has_scale
    - LOINC:has_suffix
    - LOINC:has_time_aspect
    - LOINC:has_time_modifier
    - LOINC:has_timing_of
    - NCIT:R88
    - NCIT:eo_disease_has_property_or_attribute
    - NCIT:has_data_element
    - NCIT:has_pharmaceutical_administration_method
    - NCIT:has_pharmaceutical_basic_dose_form
    - NCIT:has_pharmaceutical_intended_site
    - NCIT:has_pharmaceutical_release_characteristics
    - NCIT:has_pharmaceutical_state_of_matter
    - NCIT:has_pharmaceutical_transformation
    - NCIT:is_qualified_by
    - NCIT:qualifier_applies_to
    - NCIT:role_has_domain
    - NCIT:role_has_range
    - INO:0000154
    - HANCESTRO:0308
    - orphanet:C016
    - orphanet:C017
    - RO:0000053
    - RO:0000086
    - RO:0000087
    - SNOMED:has_access
    - SNOMED:has_clinical_course
    - SNOMED:has_count_of_base_of_active_ingredient
    - SNOMED:has_dose_form_administration_method
    - SNOMED:has_dose_form_release_characteristic
    - SNOMED:has_dose_form_transformation
    - SNOMED:has_finding_context
    - SNOMED:has_finding_informer
    - SNOMED:has_inherent_attribute
    - SNOMED:has_intent
    - SNOMED:has_interpretation
    - SNOMED:has_laterality
    - SNOMED:has_measurement_method
    - SNOMED:has_method
    - SNOMED:has_priority
    - SNOMED:has_procedure_context
    - SNOMED:has_process_duration
    - SNOMED:has_property
    - SNOMED:has_revision_status
    - SNOMED:has_scale_type
    - SNOMED:has_severity
    - SNOMED:has_specimen
    - SNOMED:has_state_of_matter
    - SNOMED:has_subject_relationship_context
    - SNOMED:has_surgical_approach
    - SNOMED:has_technique
    - SNOMED:has_temporal_context
    - SNOMED:has_time_aspect
    - SNOMED:has_units
    - UMLS:has_structural_class
    - UMLS:has_supported_concept_property
    - UMLS:has_supported_concept_relationship
    - UMLS:may_be_qualified_by
    rank: 1000
    domain: entity
    slot_uri: biolink:has_attribute
    alias: has_attribute
    owner: CellTypeTaxon
    domain_of:
    - entity
    - attribute
    - named thing
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: attribute
    multivalued: true
  deprecated:
    name: deprecated
    definition_uri: https://w3id.org/biolink/vocab/deprecated
    description: A boolean flag indicating that an entity is no longer considered
      current or valid.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    exact_mappings:
    - oboInOwl:ObsoleteClass
    rank: 1000
    slot_uri: biolink:deprecated
    alias: deprecated
    owner: CellTypeTaxon
    domain_of:
    - entity
    - attribute
    - named thing
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: boolean
  provided by:
    name: provided by
    definition_uri: https://w3id.org/biolink/vocab/provided_by
    description: The value in this node property represents the knowledge provider
      that created or assembled the node and all of its attributes.  Used internally
      to represent how a particular node made its way into a knowledge provider or
      graph.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    is_a: node property
    domain: named thing
    slot_uri: biolink:provided_by
    alias: provided_by
    owner: CellTypeTaxon
    domain_of:
    - named thing
    - attribute
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: string
    multivalued: true
  full name:
    name: full name
    definition_uri: https://w3id.org/biolink/vocab/full_name
    description: a long-form human readable name for a thing
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    is_a: node property
    domain: named thing
    slot_uri: biolink:full_name
    alias: full_name
    owner: CellTypeTaxon
    domain_of:
    - named thing
    - attribute
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: label type
  synonym:
    name: synonym
    definition_uri: https://w3id.org/biolink/vocab/synonym
    description: Alternate human-readable names for a thing
    in_subset:
    - translator_minimal
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    aliases:
    - alias
    narrow_mappings:
    - skos:altLabel
    - gff3:Alias
    - AGRKB:synonyms
    - gpi:DB_Object_Synonyms
    - HANCESTRO:0330
    - IAO:0000136
    - RXNORM:has_tradename
    rank: 1000
    is_a: node property
    domain: named thing
    slot_uri: biolink:synonym
    alias: synonym
    owner: CellTypeTaxon
    domain_of:
    - named thing
    - attribute
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: label type
    multivalued: true
  information content:
    name: information content
    definition_uri: https://w3id.org/biolink/vocab/information_content
    description: Information content (IC) value for a term, primarily from Automats.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    is_a: node property
    domain: named thing
    slot_uri: biolink:information_content
    alias: information_content
    owner: CellTypeTaxon
    domain_of:
    - named thing
    - attribute
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: float
  equivalent identifiers:
    name: equivalent identifiers
    definition_uri: https://w3id.org/biolink/vocab/equivalent_identifiers
    description: A set of identifiers that are considered equivalent to the primary
      identifier of the entity. This attribute is used to represent a collection of
      identifiers that are considered equivalent to the primary identifier of an entity.
      These equivalent identifiers may come from different databases, ontologies,
      or naming conventions, but they all refer to the same underlying concept or
      entity. This attribute is particularly useful in data integration and interoperability
      scenarios, where it is important to recognize and link different representations
      of the same entity across various sources.
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    see_also:
    - biolink:xref
    - biolink:synonyms
    rank: 1000
    is_a: node property
    domain: named thing
    slot_uri: biolink:equivalent_identifiers
    alias: equivalent_identifiers
    owner: CellTypeTaxon
    domain_of:
    - named thing
    - attribute
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    range: uriorcurie
    multivalued: true
  named thing_category:
    name: named thing_category
    definition_uri: https://w3id.org/biolink/vocab/category
    description: Name of the high level ontology class in which this entity is categorized.
      Corresponds to the label for the biolink entity type class. In a neo4j database
      this MAY correspond to the neo4j label tag. In an RDF database it should be
      a biolink model class URI. This field is multi-valued. It should include values
      for ancestors of the biolink class; for example, a protein such as Shh would
      have category values `biolink:Protein`, `biolink:GeneProduct`, `biolink:MolecularEntity`.
      In an RDF database, nodes will typically have an rdf:type triples. This can
      be to the most specific biolink class, or potentially to a class more specific
      than something in biolink. For example, a sequence feature `f` may have a rdf:type
      assertion to a SO class such as TF_binding_site, which is more specific than
      anything in biolink. Here we would have categories {biolink:GenomicEntity, biolink:MolecularEntity,
      biolink:NamedThing}
    in_subset:
    - translator_minimal
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    is_a: category
    domain: named thing
    slot_uri: biolink:category
    designates_type: true
    alias: category
    owner: CellTypeTaxon
    domain_of:
    - named thing
    - attribute
    - organism taxon
    - study result
    - relative frequency analysis result
    - information content entity
    - dataset
    - physical entity
    - activity
    - procedure
    - material sample
    - biological entity
    - gene
    - genome
    is_class_field: true
    is_usage_slot: true
    usage_slot_name: category
    range: uriorcurie
    required: true
    multivalued: true
  category:
    name: category
    definition_uri: https://w3id.org/biolink/vocab/category
    description: Name of the high level ontology class in which this entity is categorized.
      Corresponds to the label for the biolink entity type class. In a neo4j database
      this MAY correspond to the neo4j label tag. In an RDF database it should be
      a biolink model class URI. This field is multi-valued. It should include values
      for ancestors of the biolink class; for example, a protein such as Shh would
      have category values `biolink:Protein`, `biolink:GeneProduct`, `biolink:MolecularEntity`.
      In an RDF database, nodes will typically have an rdf:type triples. This can
      be to the most specific biolink class, or potentially to a class more specific
      than something in biolink. For example, a sequence feature `f` may have a rdf:type
      assertion to a SO class such as TF_binding_site, which is more specific than
      anything in biolink. Here we would have categories {biolink:GenomicEntity, biolink:MolecularEntity,
      biolink:NamedThing}
    in_subset:
    - translator_minimal
    from_schema: https://w3id.org/brain-bican/bke-taxonomy
    rank: 1000
    is_a: type
    domain: entity
    slot_uri: biolink:category
    designates_type: true
    alias: category
    owner: CellTypeTaxon
    domain_of:
    - entity
    is_class_field: true
    range: uriorcurie
    required: true
    multivalued: true

</code></pre>
</details>
</article>
</div>
<script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
</div>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
<script src="../assets/javascripts/bundle.f55a23d4.min.js"></script>
<script type="module">import mermaid from "https://unpkg.com/mermaid@10.4.0/dist/mermaid.esm.min.mjs";
mermaid.initialize({});</script></body>
</html>