
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="..">
      
      
        <link rel="next" href="../index_anatomical_structure/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.6.22">
    
    
      
        <title>Library Generation - BICAN Knowledge Graph Models Documentation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.84d31ad4.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#library-generation-schema" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="BICAN Knowledge Graph Models Documentation" class="md-header__button md-logo" aria-label="BICAN Knowledge Graph Models Documentation" data-md-component="logo">
      
  <img src="../images/bican_logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            BICAN Knowledge Graph Models Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Library Generation
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/brain-bican/models" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="BICAN Knowledge Graph Models Documentation" class="md-nav__button md-logo" aria-label="BICAN Knowledge Graph Models Documentation" data-md-component="logo">
      
  <img src="../images/bican_logo.png" alt="logo">

    </a>
    BICAN Knowledge Graph Models Documentation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/brain-bican/models" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Overview
    
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Models
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Models
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    Library Generation
    
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    Library Generation
    
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    <span class="md-ellipsis">
      Classes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      Slots
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    <span class="md-ellipsis">
      Enumerations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#types" class="md-nav__link">
    <span class="md-ellipsis">
      Types
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    <span class="md-ellipsis">
      Subsets
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../index_anatomical_structure/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Anatomical Structure
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../index_genome_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Genome Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../index_assertion_evidence/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assertion Evidence
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../index_bke_taxonomy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BKE Taxonomy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    Visualizations
    
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            Visualizations
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../library_generation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Library Generation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../anatomical_structure/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Anatomical Structure
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../genome_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Genome Annotation
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../assertion_evidence/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    Assertion Evidence
    
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bke_taxonomy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    BKE Taxonomy
    
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    <span class="md-ellipsis">
      Classes
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      Slots
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    <span class="md-ellipsis">
      Enumerations
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#types" class="md-nav__link">
    <span class="md-ellipsis">
      Types
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    <span class="md-ellipsis">
      Subsets
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="library-generation-schema">Library Generation Schema</h1>
<p>The Library Generation schema is designed to represent types and relationships of samples and digital data assets generated during processes that generate multimodal genomic data.</p>
<p>URI: https://identifiers.org/brain-bican/library-generation-schema</p>
<p>Name: library-generation-schema</p>
<h2 id="classes">Classes</h2>
<table>
<thead>
<tr>
<th>Class</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../Annotation/">Annotation</a></td>
<td>Biolink Model root class for entity annotations.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../QuantityValue/">QuantityValue</a></td>
<td>A value of an attribute that is quantitative and measurable, expressed as a combination of a unit and a numeric value</td>
</tr>
<tr>
<td><a href="../ChemicalEntityOrGeneOrGeneProduct/">ChemicalEntityOrGeneOrGeneProduct</a></td>
<td>A union of chemical entities and children, and gene or gene product. This mixin is helpful to use when searching across chemical entities that must include genes and their children as chemical entities.</td>
</tr>
<tr>
<td><a href="../Entity/">Entity</a></td>
<td>Root Biolink Model class for all things and informational relationships, real or imagined.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Checksum/">Checksum</a></td>
<td>Checksum values associated with digital entities.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../DissectionRoiPolygon/">DissectionRoiPolygon</a></td>
<td>A polygon annotated on a brain slab image delineating a region of interest (ROI) for a tissue sample dissectioning.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../NamedThing/">NamedThing</a></td>
<td>a databased entity or concept/class</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Activity/">Activity</a></td>
<td>An activity is something that occurs over a period of time and acts upon or with entities; it may include consuming, processing, transforming, modifying, relocating, using, or generating entities.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Attribute/">Attribute</a></td>
<td>A property or characteristic of an entity. For example, an apple may have properties such as color, shape, age, crispiness. An environmental sample may have attributes such as depth, lat, long, material.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../BiologicalEntity/">BiologicalEntity</a></td>
<td>None</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Gene/">Gene</a></td>
<td>A region (or regions) that includes all of the sequence elements necessary to encode a functional transcript. A gene locus may include regulatory regions, transcribed regions and/or other functional sequence regions.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Genome/">Genome</a></td>
<td>A genome is the sum of genetic material within a cell or virion.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../InformationContentEntity/">InformationContentEntity</a></td>
<td>a piece of information that typically describes some topic of discourse or is used as support.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Dataset/">Dataset</a></td>
<td>an item that refers to a collection of data from a data source.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../DigitalAsset/">DigitalAsset</a></td>
<td>None</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../StudyResult/">StudyResult</a></td>
<td>A collection of data items from a study that are about a particular study subject or experimental unit (the 'focus' of the Result) - optionally with context/provenance metadata that may be relevant to the interpretation of this data as evidence.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../RelativeFrequencyAnalysisResult/">RelativeFrequencyAnalysisResult</a></td>
<td>A result of a relative frequency analysis.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../OrganismTaxon/">OrganismTaxon</a></td>
<td>A classification of a set of organisms. Example instances: NCBITaxon:9606 (Homo sapiens), NCBITaxon:2 (Bacteria). Can also be used to represent strains or subspecies.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../PhysicalEntity/">PhysicalEntity</a></td>
<td>An entity that has material reality (a.k.a. physical essence).</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Donor/">Donor</a></td>
<td>A person or organism that is the source of a biological sample for scientific study.  Many biological samples are generated from a single donor.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../MaterialSample/">MaterialSample</a></td>
<td>A sample is a limited quantity of something (e.g. an individual or set of individuals from a population, or a portion of a substance) to be used for testing, analysis, inspection, investigation, demonstration, or trial use. [SIO]</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../AmplifiedCdna/">AmplifiedCdna</a></td>
<td>A collection of cDNA molecules derived and amplified from an input barcoded cell sample. These cDNA molecules represent the gene expression of each cell, with all cDNA molecules from a given cell retaining that cell's unique barcode from the cell barcoding step. This is a necessary step for GEX methods but is not used for ATAC methods.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../BarcodedCellSample/">BarcodedCellSample</a></td>
<td>A collection of molecularly barcoded cells. Input will be either dissociated cell sample or enriched cell sample. Cell barcodes are only guaranteed to be unique within this one collection. One dissociated cell sample or enriched cell sample can lead to multiple barcoded cell samples.  The sequences of the molecular barcodes are revealed during alignment of the resulting fastq files for the barcoded cell sample. The barcoded cell sample name and the cell level molecular barcode together uniquely identify a single cell.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../BrainSlab/">BrainSlab</a></td>
<td>A thick flat piece of brain tissue obtained by slicing a whole brain, brain hemisphere or subdivision with a blade at regular interval.  When multiple brain slabs are obtained from the slicing process, an ordinal is assigned to provide information about the relative positioning of the slabs.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../DissociatedCellSample/">DissociatedCellSample</a></td>
<td>A collection of dissociated cells or nuclei derived from dissociation of a tissue sample.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../EnrichedCellSample/">EnrichedCellSample</a></td>
<td>A collection of enriched cells or nuclei after enrichment process, usually via fluorescence-activated cell sorting (FACS) using the enrichment plan, is applied to dissociated cell sample.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Library/">Library</a></td>
<td>A collection of fragmented and barcode-indexed DNA molecules for sequencing. An index or barcode is typically introduced to enable identification of library origin to allow libraries to be pooled together for sequencing.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../LibraryAliquot/">LibraryAliquot</a></td>
<td>One library in the library pool. Each library aliquot in a library pool will have a unique R1/R2 index to allow for sequencing together then separating the sequencing output by originating library aliquot through the process of demultiplexing. The resulting demultiplexed fastq files will include the library aliquot name.  A given library may produce multiple library aliquots, which is done in the case of resequencing.  Each library aliquot will produce a set of fastq files.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../LibraryPool/">LibraryPool</a></td>
<td>A library pool is made up of library aliquots from multiple libraries. Each library aliquot in a library pool will have a unique R1/R2 index to allow for sequencing together then separating the sequencing output by originating library aliquot through the process of demultiplexing.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../TissueSample/">TissueSample</a></td>
<td>The final intact piece of tissue before cell or nuclei prep. This piece of tissue will be used in dissociation and has an region of interest polygon (ROI) associated with it.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Procedure/">Procedure</a></td>
<td>A series of actions conducted in a certain order or manner</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../CdnaAmplification/">CdnaAmplification</a></td>
<td>The process of creating a collection of cDNA molecules derived and amplified from an input barcoded cell sample.  A large amount of cDNA is needed to have accurate and reliable sequencing detection of gene expression.  This process generates multiple copies of each mRNA transcript (expressed gene) within each cell while retaining the cell's unique barcode from the barcoding step. This is a necessary step for GEX methods but is not used for ATAC methods.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../CellBarcoding/">CellBarcoding</a></td>
<td>The process of adding a molecular barcode to individual cells in a sample. The input will be either dissociated cell sample or enriched cell sample. Cell barcodes are only guaranteed to be unique within this one collection. One dissociated cell sample or enriched cell sample can lead to multiple barcoded cell samples.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../CellDissociation/">CellDissociation</a></td>
<td>The process of generating dissociated cells from an input tissue sample. This process could also introduce a tissue-source barcode (eg cell hashing), allowing mixing of cell dissociation samples at the cell barcoding step.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../CellEnrichment/">CellEnrichment</a></td>
<td>The process of enriching a dissociated cell sample by including or excluding cells of different types based on an enrichment plan using techniques such as fluorescence-activated cell sorting (FACS). This process could also introduce a tissue-source barcode (eg cell hashing), allowing mixing of cell enriched samples at the cell barcoding step.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../DissectionRoiDelineation/">DissectionRoiDelineation</a></td>
<td>The process of outlining a region of interest on a brain slab image to guide the dissection and generation of a tissue sample.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../EnrichedCellSampleSplitting/">EnrichedCellSampleSplitting</a></td>
<td>The process of splitting an enriched cell sample into several portions. Each portion may be used by the same or different groups for different scientific studies.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../LibraryConstruction/">LibraryConstruction</a></td>
<td>The process of constructing a library from input material (such as amplified cDNA or barcoded cell sample) derived from one or more cell samples.  cDNA is fragmented into smaller pieces appropriate for sequencing and at the same time a library index barcode is incorporated to enable identification of library origin, allowing libraries to be pooled together for sequencing.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../LibraryPooling/">LibraryPooling</a></td>
<td>The process of constructing of a libray pool by combining library aliquots from a set of input libraries. Each library aliquot in a library pool will have a unique R1/R2 index to allow for sequencing together then separating the sequencing output by originating library aliquot through the process of demultiplexing.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../TissueDissection/">TissueDissection</a></td>
<td>The process of dissecting a tissue sample from a brain slab guided by a dissection region of interest (ROI) delineation.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../VersionedNamedThing/">VersionedNamedThing</a></td>
<td>An iteration of the biolink:NamedThing class that stores metadata about the object's version.</td>
</tr>
<tr>
<td><a href="../GenomicEntity/">GenomicEntity</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="../MacromolecularMachineMixin/">MacromolecularMachineMixin</a></td>
<td>A union of gene locus, gene product, and macromolecular complex. These are the basic units of function in a cell. They either carry out individual biological activities, or they encode molecules which do this.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../GeneOrGeneProduct/">GeneOrGeneProduct</a></td>
<td>A union of gene loci or gene products. Frequently an identifier for one will be used as proxy for another</td>
</tr>
<tr>
<td><a href="../OntologyClass/">OntologyClass</a></td>
<td>a concept or class in an ontology, vocabulary or thesaurus. Note that nodes in a biolink compatible KG can be considered both instances of biolink classes, and OWL classes in their own right. In general you should not need to use this class directly. Instead, use the appropriate biolink class. For example, for the GO concept of endocytosis (GO:0006897), use bl:BiologicalProcess as the type.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../TaxonomicRank/">TaxonomicRank</a></td>
<td>A descriptor for the rank within a taxonomic classification. Example instance: TAXRANK:0000017 (kingdom)</td>
</tr>
<tr>
<td><a href="../PhysicalEssenceOrOccurrent/">PhysicalEssenceOrOccurrent</a></td>
<td>Either a physical or processual entity.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Occurrent/">Occurrent</a></td>
<td>A processual entity.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ActivityAndBehavior/">ActivityAndBehavior</a></td>
<td>Activity or behavior of any independent integral living, organization or mechanical actor in the world</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../PhysicalEssence/">PhysicalEssence</a></td>
<td>Semantic mixin concept.  Pertains to entities that have physical properties such as mass, volume, or charge.</td>
</tr>
<tr>
<td><a href="../ProvActivity/">ProvActivity</a></td>
<td>An activity is something that occurs over a period of time and acts upon or with entities;  it may include consuming, processing, transforming, modifying, relocating, using, or generating entities.</td>
</tr>
<tr>
<td><a href="../ProvEntity/">ProvEntity</a></td>
<td>An entity is a physical, digital, conceptual, or other kind of thing with some fixed aspects;  entities may be real or imaginary.</td>
</tr>
<tr>
<td><a href="../SubjectOfInvestigation/">SubjectOfInvestigation</a></td>
<td>An entity that has the role of being studied in an investigation, study, or experiment</td>
</tr>
<tr>
<td><a href="../ThingWithTaxon/">ThingWithTaxon</a></td>
<td>A mixin that can be used on any entity that can be taxonomically classified. This includes individual organisms; genes, their products and other molecular entities; body parts; biological processes</td>
</tr>
</tbody>
</table>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Slot</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../age_at_death_description/">age_at_death_description</a></td>
<td>Text description of the age of death following typical scientific convention ...</td>
</tr>
<tr>
<td><a href="../age_at_death_reference_point/">age_at_death_reference_point</a></td>
<td>The reference point for an age interval; for example, birth or conception</td>
</tr>
<tr>
<td><a href="../age_at_death_unit/">age_at_death_unit</a></td>
<td>The unit used for representing the donor age from the reference point</td>
</tr>
<tr>
<td><a href="../age_at_death_value/">age_at_death_value</a></td>
<td>The value representing the donor age from the reference point</td>
</tr>
<tr>
<td><a href="../amplified_cDNA_quantity_ng/">amplified_cDNA_quantity_ng</a></td>
<td>Amount of cDNA produced after cDNA amplification measured in nanograms</td>
</tr>
<tr>
<td><a href="../amplified_cDNA_result/">amplified_cDNA_result</a></td>
<td>Pass or Fail result based on qualitative assessment of cDNA yield and size</td>
</tr>
<tr>
<td><a href="../annotates/">annotates</a></td>
<td>The brain slab that was annotated by the delineation process</td>
</tr>
<tr>
<td><a href="../attribute_name/">attribute_name</a></td>
<td>The human-readable 'attribute name' can be set to a string which reflects its...</td>
</tr>
<tr>
<td><a href="../average_size_bp/">average_size_bp</a></td>
<td>Average size of the library in terms of base pairs</td>
</tr>
<tr>
<td><a href="../biological_sex/">biological_sex</a></td>
<td>Biological sex of donor at birth</td>
</tr>
<tr>
<td><a href="../category/">category</a></td>
<td>Name of the high level ontology class in which this entity is categorized</td>
</tr>
<tr>
<td><a href="../cDNA_amplification_process_date/">cDNA_amplification_process_date</a></td>
<td>Date of cDNA amplification</td>
</tr>
<tr>
<td><a href="../cDNA_amplification_set/">cDNA_amplification_set</a></td>
<td>cDNA amplification set, containing multiple amplified_cDNA_names that were pr...</td>
</tr>
<tr>
<td><a href="../cell_barcoding_method/">cell_barcoding_method</a></td>
<td>Standardized nomenclature to describe the general barcoding method used</td>
</tr>
<tr>
<td><a href="../cell_barcoding_process_date/">cell_barcoding_process_date</a></td>
<td>Date of cell barcoding process</td>
</tr>
<tr>
<td><a href="../cell_dissociation_process_date/">cell_dissociation_process_date</a></td>
<td>Date of cell dissociation process</td>
</tr>
<tr>
<td><a href="../cell_enrichment_process_date/">cell_enrichment_process_date</a></td>
<td>Date of cell enrichment process</td>
</tr>
<tr>
<td><a href="../cell_prep_type/">cell_prep_type</a></td>
<td>The type of cell preparation</td>
</tr>
<tr>
<td><a href="../checksum_algorithm/">checksum_algorithm</a></td>
<td>The type of cryptographic hash function used to calculate the checksum value</td>
</tr>
<tr>
<td><a href="../chembl_chirality/">chembl_chirality</a></td>
<td>Tern indicating the chirality of the chemical entity</td>
</tr>
<tr>
<td><a href="../chembl_drug_warning/">chembl_drug_warning</a></td>
<td>Text describing warnings for use of chemicals as therapeutics</td>
</tr>
<tr>
<td><a href="../chembl_prodrug/">chembl_prodrug</a></td>
<td>Flag indicating if a drug is a prodrug that is active only after being metabo...</td>
</tr>
<tr>
<td><a href="../concentration_nm/">concentration_nm</a></td>
<td>Concentration of library in terms of nM (nMol/L)</td>
</tr>
<tr>
<td><a href="../content_url/">content_url</a></td>
<td></td>
</tr>
<tr>
<td><a href="../creation_date/">creation_date</a></td>
<td>date on which an entity was created</td>
</tr>
<tr>
<td><a href="../data_type/">data_type</a></td>
<td>The type of data in the file</td>
</tr>
<tr>
<td><a href="../deprecated/">deprecated</a></td>
<td>A boolean flag indicating that an entity is no longer considered current or v...</td>
</tr>
<tr>
<td><a href="../description/">description</a></td>
<td>a human-readable description of an entity</td>
</tr>
<tr>
<td><a href="../digest/">digest</a></td>
<td>Stores checksum information</td>
</tr>
<tr>
<td><a href="../dissection_was_guided_by/">dissection_was_guided_by</a></td>
<td>The dissection ROI polygon that was used to guide the dissection</td>
</tr>
<tr>
<td><a href="../dissociated_cell_oligo_name/">dissociated_cell_oligo_name</a></td>
<td>Name of cell source oligo used in cell plexing</td>
</tr>
<tr>
<td><a href="../enriched_cell_oligo_name/">enriched_cell_oligo_name</a></td>
<td>Name of cell source oligo used in cell plexing</td>
</tr>
<tr>
<td><a href="../enrichment_population/">enrichment_population</a></td>
<td>Actual percentage of cells as a result of using set of fluorescent marker lab...</td>
</tr>
<tr>
<td><a href="../equivalent_identifiers/">equivalent_identifiers</a></td>
<td>A set of identifiers that are considered equivalent to the primary identifier...</td>
</tr>
<tr>
<td><a href="../format/">format</a></td>
<td></td>
</tr>
<tr>
<td><a href="../full_name/">full_name</a></td>
<td>a long-form human readable name for a thing</td>
</tr>
<tr>
<td><a href="../has_attribute/">has_attribute</a></td>
<td>connects any entity to an attribute</td>
</tr>
<tr>
<td><a href="../has_attribute_type/">has_attribute_type</a></td>
<td>connects an attribute to a class that describes it</td>
</tr>
<tr>
<td><a href="../has_biological_sequence/">has_biological_sequence</a></td>
<td>connects a genomic feature to its sequence</td>
</tr>
<tr>
<td><a href="../has_numeric_value/">has_numeric_value</a></td>
<td>connects a quantity value to a number</td>
</tr>
<tr>
<td><a href="../has_qualitative_value/">has_qualitative_value</a></td>
<td>connects an attribute to a value</td>
</tr>
<tr>
<td><a href="../has_quantitative_value/">has_quantitative_value</a></td>
<td>connects an attribute to a value</td>
</tr>
<tr>
<td><a href="../has_taxonomic_rank/">has_taxonomic_rank</a></td>
<td></td>
</tr>
<tr>
<td><a href="../has_unit/">has_unit</a></td>
<td>connects a quantity value to a unit</td>
</tr>
<tr>
<td><a href="../histone_modification_marker/">histone_modification_marker</a></td>
<td>Histone modification marker antibodies (eg H3K27ac, H3K27me3, H3K9me3) used i...</td>
</tr>
<tr>
<td><a href="../id/">id</a></td>
<td>A unique identifier for an entity</td>
</tr>
<tr>
<td><a href="../in_taxon/">in_taxon</a></td>
<td>connects an entity to its taxonomic classification</td>
</tr>
<tr>
<td><a href="../in_taxon_label/">in_taxon_label</a></td>
<td>The human readable scientific name for the taxon of the entity</td>
</tr>
<tr>
<td><a href="../information_content/">information_content</a></td>
<td>Information content (IC) value for a term, primarily from Automats</td>
</tr>
<tr>
<td><a href="../input_quantity/">input_quantity</a></td>
<td>Number of enriched or dissociated cells/nuclei going into the barcoding proce...</td>
</tr>
<tr>
<td><a href="../input_quantity_ng/">input_quantity_ng</a></td>
<td>Amount of cDNA going into library construction in nanograms</td>
</tr>
<tr>
<td><a href="../iri/">iri</a></td>
<td>An IRI for an entity</td>
</tr>
<tr>
<td><a href="../library_construction_method/">library_construction_method</a></td>
<td>Standardized nomenclature to describe the specific library method used</td>
</tr>
<tr>
<td><a href="../library_construction_process_date/">library_construction_process_date</a></td>
<td>Date of library construction</td>
</tr>
<tr>
<td><a href="../library_construction_set/">library_construction_set</a></td>
<td>Library set, containing multiple library_names that were processed at the sam...</td>
</tr>
<tr>
<td><a href="../library_quantity_ng/">library_quantity_ng</a></td>
<td>Amount of library generated in terms of nanograms</td>
</tr>
<tr>
<td><a href="../library_result/">library_result</a></td>
<td>Pass or Fail result based on qualitative assessment of library yield and size</td>
</tr>
<tr>
<td><a href="../license/">license</a></td>
<td></td>
</tr>
<tr>
<td><a href="../local_tube_id/">local_tube_id</a></td>
<td>Library Pool Tube local name</td>
</tr>
<tr>
<td><a href="../macromolecular_machine_mixin_name/">macromolecular_machine_mixin_name</a></td>
<td>genes are typically designated by a short symbol and a full name</td>
</tr>
<tr>
<td><a href="../name/">name</a></td>
<td>A human-readable name for an attribute or entity</td>
</tr>
<tr>
<td><a href="../named_thing_category/">named_thing_category</a></td>
<td>Name of the high level ontology class in which this entity is categorized</td>
</tr>
<tr>
<td><a href="../node_property/">node_property</a></td>
<td>A grouping for any property that holds between a node and a value</td>
</tr>
<tr>
<td><a href="../number_of_expected_cells/">number_of_expected_cells</a></td>
<td>Expected number of cells/nuclei of a barcoded_cell_sample that will be barcod...</td>
</tr>
<tr>
<td><a href="../organism_taxon_has_taxonomic_rank/">organism_taxon_has_taxonomic_rank</a></td>
<td></td>
</tr>
<tr>
<td><a href="../pcr_cycles/">pcr_cycles</a></td>
<td>Number of PCR cycles used during cDNA amplification for this cDNA</td>
</tr>
<tr>
<td><a href="../percent_cdna_longer_than_400bp/">percent_cdna_longer_than_400bp</a></td>
<td>QC metric to measure mRNA degradation of cDNA</td>
</tr>
<tr>
<td><a href="../port_well/">port_well</a></td>
<td>Specific position of the loaded port of the 10x chip</td>
</tr>
<tr>
<td><a href="../process_date/">process_date</a></td>
<td>Date of library pooling process</td>
</tr>
<tr>
<td><a href="../provided_by/">provided_by</a></td>
<td>The value in this node property represents the knowledge provider that create...</td>
</tr>
<tr>
<td><a href="../quantity_fmol/">quantity_fmol</a></td>
<td>Amount of library generated in terms of femtomoles</td>
</tr>
<tr>
<td><a href="../r1_r2_index/">r1_r2_index</a></td>
<td>Name of the pair of library indexes used for sequencing</td>
</tr>
<tr>
<td><a href="../related_to/">related_to</a></td>
<td>A relationship that is asserted between two named things</td>
</tr>
<tr>
<td><a href="../related_to_at_instance_level/">related_to_at_instance_level</a></td>
<td>Represents a relationship held between two instances of a data classes</td>
</tr>
<tr>
<td><a href="../revision_of/">revision_of</a></td>
<td></td>
</tr>
<tr>
<td><a href="../rights/">rights</a></td>
<td></td>
</tr>
<tr>
<td><a href="../species/">species</a></td>
<td>Species of donor</td>
</tr>
<tr>
<td><a href="../structure/">structure</a></td>
<td>Structure of tissue sample</td>
</tr>
<tr>
<td><a href="../symbol/">symbol</a></td>
<td>Symbol for a particular thing</td>
</tr>
<tr>
<td><a href="../synonym/">synonym</a></td>
<td>Alternate human-readable names for a thing</td>
</tr>
<tr>
<td><a href="../type/">type</a></td>
<td></td>
</tr>
<tr>
<td><a href="../used/">used</a></td>
<td>Usage is the beginning of utilizing an entity by an activity</td>
</tr>
<tr>
<td><a href="../value/">value</a></td>
<td>The checksum value obtained from a specific cryotographic hash function</td>
</tr>
<tr>
<td><a href="../version/">version</a></td>
<td></td>
</tr>
<tr>
<td><a href="../was_derived_from/">was_derived_from</a></td>
<td>A derivation is a transformation of an entity into another, an update of an e...</td>
</tr>
<tr>
<td><a href="../was_generated_by/">was_generated_by</a></td>
<td>Generation is the completion of production of a new entity by an activity</td>
</tr>
<tr>
<td><a href="../was_guided_by/">was_guided_by</a></td>
<td>The dissection ROI polygon which was used to guide the tissue dissection</td>
</tr>
<tr>
<td><a href="../xref/">xref</a></td>
<td>A database cross reference or alternative identifier for a NamedThing or edge...</td>
</tr>
</tbody>
</table>
<h2 id="enumerations">Enumerations</h2>
<table>
<thead>
<tr>
<th>Enumeration</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../AgeAtDeathReferencePoint/">AgeAtDeathReferencePoint</a></td>
<td></td>
</tr>
<tr>
<td><a href="../AgeAtDeathUnit/">AgeAtDeathUnit</a></td>
<td></td>
</tr>
<tr>
<td><a href="../AmplifiedCdnaRnaAmplificationPassFail/">AmplifiedCdnaRnaAmplificationPassFail</a></td>
<td></td>
</tr>
<tr>
<td><a href="../BarcodedCellSampleTechnique/">BarcodedCellSampleTechnique</a></td>
<td></td>
</tr>
<tr>
<td><a href="../DigestType/">DigestType</a></td>
<td></td>
</tr>
<tr>
<td><a href="../DissociatedCellSampleCellLabelBarcode/">DissociatedCellSampleCellLabelBarcode</a></td>
<td></td>
</tr>
<tr>
<td><a href="../DissociatedCellSampleCellPrepType/">DissociatedCellSampleCellPrepType</a></td>
<td></td>
</tr>
<tr>
<td><a href="../LibraryPrepPassFail/">LibraryPrepPassFail</a></td>
<td></td>
</tr>
<tr>
<td><a href="../LibraryR1R2Index/">LibraryR1R2Index</a></td>
<td></td>
</tr>
<tr>
<td><a href="../LibraryTechnique/">LibraryTechnique</a></td>
<td></td>
</tr>
<tr>
<td><a href="../Sex/">Sex</a></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="types">Types</h2>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../BiologicalSequence/">BiologicalSequence</a></td>
<td></td>
</tr>
<tr>
<td><a href="../Boolean/">Boolean</a></td>
<td>A binary (true or false) value</td>
</tr>
<tr>
<td><a href="../Curie/">Curie</a></td>
<td>a compact URI</td>
</tr>
<tr>
<td><a href="../Date/">Date</a></td>
<td>a date (year, month and day) in an idealized calendar</td>
</tr>
<tr>
<td><a href="../DateOrDatetime/">DateOrDatetime</a></td>
<td>Either a date or a datetime</td>
</tr>
<tr>
<td><a href="../Datetime/">Datetime</a></td>
<td>The combination of a date and time</td>
</tr>
<tr>
<td><a href="../Decimal/">Decimal</a></td>
<td>A real number with arbitrary precision that conforms to the xsd:decimal speci...</td>
</tr>
<tr>
<td><a href="../Double/">Double</a></td>
<td>A real number that conforms to the xsd:double specification</td>
</tr>
<tr>
<td><a href="../Float/">Float</a></td>
<td>A real number that conforms to the xsd:float specification</td>
</tr>
<tr>
<td><a href="../Integer/">Integer</a></td>
<td>An integer</td>
</tr>
<tr>
<td><a href="../IriType/">IriType</a></td>
<td>An IRI</td>
</tr>
<tr>
<td><a href="../Jsonpath/">Jsonpath</a></td>
<td>A string encoding a JSON Path</td>
</tr>
<tr>
<td><a href="../Jsonpointer/">Jsonpointer</a></td>
<td>A string encoding a JSON Pointer</td>
</tr>
<tr>
<td><a href="../LabelType/">LabelType</a></td>
<td>A string that provides a human-readable name for an entity</td>
</tr>
<tr>
<td><a href="../NarrativeText/">NarrativeText</a></td>
<td>A string that provides a human-readable description of something</td>
</tr>
<tr>
<td><a href="../Ncname/">Ncname</a></td>
<td>Prefix part of CURIE</td>
</tr>
<tr>
<td><a href="../Nodeidentifier/">Nodeidentifier</a></td>
<td>A URI, CURIE or BNODE that represents a node in a model</td>
</tr>
<tr>
<td><a href="../Objectidentifier/">Objectidentifier</a></td>
<td>A URI or CURIE that represents an object in the model</td>
</tr>
<tr>
<td><a href="../Sparqlpath/">Sparqlpath</a></td>
<td>A string encoding a SPARQL Property Path</td>
</tr>
<tr>
<td><a href="../String/">String</a></td>
<td>A character string</td>
</tr>
<tr>
<td><a href="../SymbolType/">SymbolType</a></td>
<td></td>
</tr>
<tr>
<td><a href="../Time/">Time</a></td>
<td>A time object represents a (local) time of day, independent of any particular...</td>
</tr>
<tr>
<td><a href="../Unit/">Unit</a></td>
<td></td>
</tr>
<tr>
<td><a href="../Uri/">Uri</a></td>
<td>a complete URI</td>
</tr>
<tr>
<td><a href="../Uriorcurie/">Uriorcurie</a></td>
<td>a URI or a CURIE</td>
</tr>
</tbody>
</table>
<h2 id="subsets">Subsets</h2>
<table>
<thead>
<tr>
<th>Subset</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../Alignment/">Alignment</a></td>
<td>A subset of slots/attributes that are required for alignment</td>
</tr>
<tr>
<td><a href="../Analysis/">Analysis</a></td>
<td>A subset of slots/attributes that are required for analysis</td>
</tr>
<tr>
<td><a href="../Bican/">Bican</a></td>
<td>A subset of classes that are associated with BICAN</td>
</tr>
<tr>
<td><a href="../LibraryGeneration/">LibraryGeneration</a></td>
<td>A subset of classes that are associated with library generation</td>
</tr>
<tr>
<td><a href="../ModelOrganismDatabase/">ModelOrganismDatabase</a></td>
<td>Subset that is relevant for a typical Model Organism Database (MOD)</td>
</tr>
<tr>
<td><a href="../ProcessingElements/">ProcessingElements</a></td>
<td>A subset of classes that are associated with processing</td>
</tr>
<tr>
<td><a href="../Samples/">Samples</a></td>
<td>Sample/biosample datamodel</td>
</tr>
<tr>
<td><a href="../SequencingElements/">SequencingElements</a></td>
<td>A subset of classes that are associated with sequencing</td>
</tr>
<tr>
<td><a href="../Testing/">Testing</a></td>
<td>TBD</td>
</tr>
<tr>
<td><a href="../TissueSpecimen/">TissueSpecimen</a></td>
<td>A subset of classes that are associated with tissue specimens</td>
</tr>
<tr>
<td><a href="../Tracking/">Tracking</a></td>
<td>A subset of slots/attributes that are required for tracking</td>
</tr>
<tr>
<td><a href="../TranslatorMinimal/">TranslatorMinimal</a></td>
<td>Minimum subset of translator work</td>
</tr>
</tbody>
</table>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      <script id="__config" type="application/json">{"base": "..", "features": ["content.tabs.link"], "search": "../assets/javascripts/workers/search.973d3a69.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.f55a23d4.min.js"></script>
      
    
  </body>
</html>