
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="prev" href="../index_assertion_evidence/">
      
      
        <link rel="next" href="../library_generation/">
      
      
        
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.1, mkdocs-material-9.7.1">
    
    
      
        <title>BKE Taxonomy - BICAN Knowledge Graph Models Documentation</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.484c7ddc.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ab4e12ef.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce(((e,_)=>(e<<5)-e+_.charCodeAt(0)),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="white" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#bke-taxonomy-schema" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="BICAN Knowledge Graph Models Documentation" class="md-header__button md-logo" aria-label="BICAN Knowledge Graph Models Documentation" data-md-component="logo">
      
  <img src="../images/bican_logo.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3zm0 5h18v2H3zm0 5h18v2H3z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            BICAN Knowledge Graph Models Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              BKE Taxonomy
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      
      
        <label class="md-header__button md-icon" for="__search">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        </label>
        <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.52 6.52 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
      
    
    
      <div class="md-header__source">
        <a href="https://github.com/brain-bican/models" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="BICAN Knowledge Graph Models Documentation" class="md-nav__button md-logo" aria-label="BICAN Knowledge Graph Models Documentation" data-md-component="logo">
      
  <img src="../images/bican_logo.png" alt="logo">

    </a>
    BICAN Knowledge Graph Models Documentation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/brain-bican/models" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 7.1.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2025 Fonticons, Inc.--><path d="M439.6 236.1 244 40.5c-5.4-5.5-12.8-8.5-20.4-8.5s-15 3-20.4 8.4L162.5 81l51.5 51.5c27.1-9.1 52.7 16.8 43.4 43.7l49.7 49.7c34.2-11.8 61.2 31 35.5 56.7-26.5 26.5-70.2-2.9-56-37.3L240.3 199v121.9c25.3 12.5 22.3 41.8 9.1 55-6.4 6.4-15.2 10.1-24.3 10.1s-17.8-3.6-24.3-10.1c-17.6-17.6-11.1-46.9 11.2-56v-123c-20.8-8.5-24.6-30.7-18.6-45L142.6 101 8.5 235.1C3 240.6 0 247.9 0 255.5s3 15 8.5 20.4l195.6 195.7c5.4 5.4 12.7 8.4 20.4 8.4s15-3 20.4-8.4l194.7-194.7c5.4-5.4 8.4-12.8 8.4-20.4s-3-15-8.4-20.4"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Overview
  

    
  </span>
  
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" checked>
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Models
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            
  
    Models
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../index_library_generation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Library Generation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../index_anatomical_structure/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Anatomical Structure
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../index_genome_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Genome Annotation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../index_assertion_evidence/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Assertion Evidence
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  
  <span class="md-ellipsis">
    
  
    BKE Taxonomy
  

    
  </span>
  
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  
  <span class="md-ellipsis">
    
  
    BKE Taxonomy
  

    
  </span>
  
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Classes
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      
        Slots
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Enumerations
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#types" class="md-nav__link">
    <span class="md-ellipsis">
      
        Types
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    <span class="md-ellipsis">
      
        Subsets
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_3" >
        
          
          <label class="md-nav__link" for="__nav_3" id="__nav_3_label" tabindex="0">
            
  
  
  <span class="md-ellipsis">
    
  
    Visualizations
  

    
  </span>
  
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_3_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_3">
            <span class="md-nav__icon md-icon"></span>
            
  
    Visualizations
  

          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../library_generation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Library Generation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../anatomical_structure/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Anatomical Structure
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../genome_annotation/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Genome Annotation
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../assertion_evidence/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    Assertion Evidence
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../bke_taxonomy/" class="md-nav__link">
        
  
  
  <span class="md-ellipsis">
    
  
    BKE Taxonomy
  

    
  </span>
  
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#classes" class="md-nav__link">
    <span class="md-ellipsis">
      
        Classes
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#slots" class="md-nav__link">
    <span class="md-ellipsis">
      
        Slots
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#enumerations" class="md-nav__link">
    <span class="md-ellipsis">
      
        Enumerations
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#types" class="md-nav__link">
    <span class="md-ellipsis">
      
        Types
      
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#subsets" class="md-nav__link">
    <span class="md-ellipsis">
      
        Subsets
      
    </span>
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              
              <article class="md-content__inner md-typeset">
                
                  


  
  


<h1 id="bke-taxonomy-schema">BKE Taxonomy Schema</h1>
<p>The BKE Taxonomy Schema represents the structure of classifications of cell types and their hierarchical relationships in the mammalian brain.</p>
<p>URI: https://w3id.org/brain-bican/bke-taxonomy</p>
<p>Name: BKETaxonomy</p>
<h2 id="classes">Classes</h2>
<table>
<thead>
<tr>
<th>Class</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../Annotation/">Annotation</a></td>
<td>Biolink Model root class for entity annotations.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../QuantityValue/">QuantityValue</a></td>
<td>A value of an attribute that is quantitative and measurable, expressed as a combination of a unit and a numeric value</td>
</tr>
<tr>
<td><a href="../AnnotationCollection/">AnnotationCollection</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="../ChemicalEntityOrGeneOrGeneProduct/">ChemicalEntityOrGeneOrGeneProduct</a></td>
<td>A union of chemical entities and children, and gene or gene product. This mixin is helpful to use when searching across chemical entities that must include genes and their children as chemical entities.</td>
</tr>
<tr>
<td><a href="../Entity/">Entity</a></td>
<td>Root Biolink Model class for all things and informational relationships, real or imagined.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Checksum/">Checksum</a></td>
<td>Checksum values associated with digital entities.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../NamedThing/">NamedThing</a></td>
<td>a databased entity or concept/class</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Abbreviation/">Abbreviation</a></td>
<td>Maps an abbreviation string to its decoded meaning. Optionally the abbreviation can be link to one or more entities it denotes.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Activity/">Activity</a></td>
<td>An activity is something that occurs over a period of time and acts upon or with entities; it may include consuming, processing, transforming, modifying, relocating, using, or generating entities.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Attribute/">Attribute</a></td>
<td>A property or characteristic of an entity. For example, an apple may have properties such as color, shape, age, crispiness. An environmental sample may have attributes such as depth, lat, long, material.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../BiologicalEntity/">BiologicalEntity</a></td>
<td>None</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Gene/">Gene</a></td>
<td>A region (or regions) that includes all of the sequence elements necessary to encode a functional transcript. A gene locus may include regulatory regions, transcribed regions and/or other functional sequence regions.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../GeneAnnotation/">GeneAnnotation</a></td>
<td>Represents a single gene. Includes metadata about the gene, such as its molecular type and the genome annotation it was referenced from.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Genome/">Genome</a></td>
<td>A genome is the sum of genetic material within a cell or virion.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../GenomeAnnotation/">GenomeAnnotation</a></td>
<td>Represents a genome annotation. Includes metadata about the genome, such as its version and reference assembly.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../CellSpecimen/">CellSpecimen</a></td>
<td>( defined as in BERS)</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../CellTypeSet/">CellTypeSet</a></td>
<td>A grouping of cell type taxons. The grouping may represent a specific taxonomic level (rank) or cell type neighborhood.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../CellTypeTaxon/">CellTypeTaxon</a></td>
<td>A cell type taxon is a node in a taxonomy. Taxons represents a unit of cell type classification and is defined by a set of clusters. Taxons may be organized into a heirarchy and levels. Taxons at higher levels represents broader cell type classes and partitioned into more specific types at lower levels. Additionally, taxons may be organized into neighborhoods or related groups.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../CellTypeTaxonomy/">CellTypeTaxonomy</a></td>
<td>A cell type taxonomy is a systematic classification of cell types and their heirarchical relationships and groupings. A taxonomy may include concepts such as heirarchical taxonomic levels or (potentially overlapping) neighborhoods.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Cluster/">Cluster</a></td>
<td>One specific cluster from the set of clusters determined by a clustering process. A cluster is defined by set of observations which has been grouped together based on similarities or patterns.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ClusterSet/">ClusterSet</a></td>
<td>The output of a clustering process where input observations are organized or classified into a set of clusters based on similarities or patterns.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ColorPalette/">ColorPalette</a></td>
<td>A schematic set of display colors that can be applied to individual components of the associated entity set (for example, a taxonomy).</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../DisplayColor/">DisplayColor</a></td>
<td>One element of a color palette representing the association between a color and entity,</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../GenomeAssembly/">GenomeAssembly</a></td>
<td>Represents a genome assembly. A genome assembly is a computational representation of a genome sequence.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../InformationContentEntity/">InformationContentEntity</a></td>
<td>a piece of information that typically describes some topic of discourse or is used as support.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Dataset/">Dataset</a></td>
<td>an item that refers to a collection of data from a data source.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../StudyResult/">StudyResult</a></td>
<td>A collection of data items from a study that are about a particular study subject or experimental unit (the 'focus' of the Result) - optionally with context/provenance metadata that may be relevant to the interpretation of this data as evidence.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../RelativeFrequencyAnalysisResult/">RelativeFrequencyAnalysisResult</a></td>
<td>A result of a relative frequency analysis.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../SpatialProportions/">SpatialProportions</a></td>
<td>Spatial distribution of expression, labeling, or anatomical density of a given entity — typically a cell type, cluster, or signal source — across named brain regions.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../MatrixFile/">MatrixFile</a></td>
<td>A digital file containing the representation of full or partial data matrix.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ObservationMatrix/">ObservationMatrix</a></td>
<td>A matrix (potentially virtual) of observations. Each row represents a specific observation or measurements on specific sample (such as a cell) over a set of variables or features. Each column represents the set of measurements (such as gene expression) of a specific variable (such as gene) over all samples.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ObservationRow/">ObservationRow</a></td>
<td>One specific row of the observation matrix representing a set of measurements preformed on a sample over a set of variables or features.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../OrganismTaxon/">OrganismTaxon</a></td>
<td>A classification of a set of organisms. Example instances: NCBITaxon:9606 (Homo sapiens), NCBITaxon:2 (Bacteria). Can also be used to represent strains or subspecies.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../PhysicalEntity/">PhysicalEntity</a></td>
<td>An entity that has material reality (a.k.a. physical essence).</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../MaterialSample/">MaterialSample</a></td>
<td>A sample is a limited quantity of something (e.g. an individual or set of individuals from a population, or a portion of a substance) to be used for testing, analysis, inspection, investigation, demonstration, or trial use. [SIO]</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Procedure/">Procedure</a></td>
<td>A series of actions conducted in a certain order or manner</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../CellTypeTaxonomyCreationProcess/">CellTypeTaxonomyCreationProcess</a></td>
<td>The process of organizing cells or clusters into a systematic classification of cell types and their heirarchical relationships and groupings.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ClusteringProcess/">ClusteringProcess</a></td>
<td>The process of organizing and classifying observations into groups or clusters based on similarities or patterns.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ObservationMatrixCreationProcess/">ObservationMatrixCreationProcess</a></td>
<td>The process of aggregating data to create an observation matrix (potentially virtual) for the purpose of dissemination or performing an analysis (such as clustering).</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../VersionedNamedThing/">VersionedNamedThing</a></td>
<td>An iteration of the biolink:NamedThing class that stores metadata about the object's version.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../AnatomicalAnnotationSet/">AnatomicalAnnotationSet</a></td>
<td>An anatomical annotation set is a versioned release of a set of anatomical annotations anchored  in the same anatomical space that divides the space into distinct segments following some annotation  criteria or parcellation scheme. For example, the anatomical annotation set of 3D image based  reference atlases (e.g. Allen Mouse CCF) can be expressed as a set of label indices of single  multi-valued image annotations or as a set of segmentation masks (ref: ILX:0777108, RRID:SCR_023499)</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../AnatomicalSpace/">AnatomicalSpace</a></td>
<td>An anatomical space is versioned release of a mathematical space with a defined mapping  between the anatomical axes and the mathematical axes. An anatomical space may be defined by  a reference image chosen as the biological reference for an anatomical structure of interest  derived from a single or multiple specimens (ref: ILX:0777106, RRID:SCR_023499)</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ImageDataset/">ImageDataset</a></td>
<td>An image dataset is versioned release of a multidimensional regular grid of measurements  and metadata required for a morphological representation of an entity such as an anatomical  structure (ref: OBI_0003327, RRID:SCR_006266)</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ParcellationAtlas/">ParcellationAtlas</a></td>
<td>A parcellation atlas is a versioned release reference used to guide experiments or deal with the spatial relationship between  objects or the location of objects within the context of some anatomical structure. An atlas is minimally defined by a notion  of space (either implicit or explicit) and an annotation set. Reference atlases usually have additional parts that make them  more useful in certain situations, such as a well defined coordinate system, delineations indicating the boundaries of various  regions or cell populations, landmarks, and labels and names to make it easier to communicate about well known and useful  locations (ref: ILX:0777109, RRID:SCR_023499).</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ParcellationColorScheme/">ParcellationColorScheme</a></td>
<td>A parcellation color scheme is a versioned release color palette that can be used to visualize a  parcellation terminology or its related parcellation annotation. A parcellation terminology may  have zero or more parcellation color schemes and each color scheme is in context of a specific  parcellation terminology, where each parcellation term is assigned a hex color value. A parcellation  color scheme is defined as a part of one and only one parcellation terminology.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ParcellationTerm/">ParcellationTerm</a></td>
<td>A parcellation term is an individual term within a specific parcellation terminology describing a  single anatomical entity by a persistent identifier, name, symbol and description.  A parcellation  term is a unique and exclusive member of a versioned release parcellation terminology. Although term  identifiers must be unique within the context of one versioned release of a parcellation terminology,  they can be reused in different parcellation terminology versions enabling the representation of  terminology updates and modifications over time.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ParcellationTerminology/">ParcellationTerminology</a></td>
<td>A parcellation terminology is a versioned release set of terms that can be used to label  annotations in an atlas, providing human readability and context and allowing communication  about brain locations and structural properties. Typically, a terminology is a set of  descriptive anatomical terms following a specific naming convention and/or approach to  organization scheme. The terminology may be a flat list of controlled vocabulary, a taxonomy  and partonomy, or an ontology (ref: ILX:0777107, RRID:SCR_023499)</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ParcellationTermSet/">ParcellationTermSet</a></td>
<td>A parcellation term set is the set of parcellation terms within a specific parcellation terminology.  A parcellation term set belongs to one and only one parcellation terminology and each parcellation  term in a parcellation terminology belongs to one and only one term set.  If the parcellation terminology is a taxonomy, parcellation term sets can be used to represent  taxonomic ranks. For consistency, if the terminology does not have the notion of taxonomic ranks,  all terms are grouped into a single parcellation term set.</td>
</tr>
<tr>
<td><a href="../GenomicEntity/">GenomicEntity</a></td>
<td>None</td>
</tr>
<tr>
<td><a href="../MacromolecularMachineMixin/">MacromolecularMachineMixin</a></td>
<td>A union of gene locus, gene product, and macromolecular complex. These are the basic units of function in a cell. They either carry out individual biological activities, or they encode molecules which do this.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../GeneOrGeneProduct/">GeneOrGeneProduct</a></td>
<td>A union of gene loci or gene products. Frequently an identifier for one will be used as proxy for another</td>
</tr>
<tr>
<td><a href="../OntologyClass/">OntologyClass</a></td>
<td>a concept or class in an ontology, vocabulary or thesaurus. Note that nodes in a biolink compatible KG can be considered both instances of biolink classes, and OWL classes in their own right. In general you should not need to use this class directly. Instead, use the appropriate biolink class. For example, for the GO concept of endocytosis (GO:0006897), use bl:BiologicalProcess as the type.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../TaxonomicRank/">TaxonomicRank</a></td>
<td>A descriptor for the rank within a taxonomic classification. Example instance: TAXRANK:0000017 (kingdom)</td>
</tr>
<tr>
<td><a href="../ParcellationAnnotation/">ParcellationAnnotation</a></td>
<td>A parcellation annotation defines a specific segment of an anatomical space denoted by an internal  identifier and is a unique and exclusive member of a versioned release anatomical annotation set.  For example, in the case where the anatomical annotation set is a single multi-value image mask (e.g. Allen Mouse CCF), a specific annotation corresponds to a specific label index (internal identifier) in the mask.</td>
</tr>
<tr>
<td><a href="../ParcellationAnnotationTermMap/">ParcellationAnnotationTermMap</a></td>
<td>The parcellation annotation term map table defines the relationship between parcellation annotations and parcellation terms.  A parcellation term is uniquely denoted by a parcellation term identifier and the parcellation terminology it belongs to.  A parcellation term can be spatially parameterized by the union of one or more parcellation annotations within a versioned  release of an anatomical annotation set. For example, annotations defining individual cortical layers in cortical region  R (R1, R2/3, R4, etc) can be combined to define the parent region R.</td>
</tr>
<tr>
<td><a href="../ParcellationColorAssignment/">ParcellationColorAssignment</a></td>
<td>The parcellation color assignment associates hex color value to a parcellation term within a  versioned release of a color scheme. A parcellation term is uniquely denoted by a parcellation  term identifier and the parcellation terminology it belongs to.</td>
</tr>
<tr>
<td><a href="../PhysicalEssenceOrOccurrent/">PhysicalEssenceOrOccurrent</a></td>
<td>Either a physical or processual entity.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../Occurrent/">Occurrent</a></td>
<td>A processual entity.</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../ActivityAndBehavior/">ActivityAndBehavior</a></td>
<td>Activity or behavior of any independent integral living, organization or mechanical actor in the world</td>
</tr>
<tr>
<td>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<a href="../PhysicalEssence/">PhysicalEssence</a></td>
<td>Semantic mixin concept.  Pertains to entities that have physical properties such as mass, volume, or charge.</td>
</tr>
<tr>
<td><a href="../ProvActivity/">ProvActivity</a></td>
<td>An activity is something that occurs over a period of time and acts upon or with entities;  it may include consuming, processing, transforming, modifying, relocating, using, or generating entities.</td>
</tr>
<tr>
<td><a href="../ProvEntity/">ProvEntity</a></td>
<td>An entity is a physical, digital, conceptual, or other kind of thing with some fixed aspects;  entities may be real or imaginary.</td>
</tr>
<tr>
<td><a href="../SubjectOfInvestigation/">SubjectOfInvestigation</a></td>
<td>An entity that has the role of being studied in an investigation, study, or experiment</td>
</tr>
<tr>
<td><a href="../ThingWithTaxon/">ThingWithTaxon</a></td>
<td>A mixin that can be used on any entity that can be taxonomically classified. This includes individual organisms; genes, their products and other molecular entities; body parts; biological processes</td>
</tr>
</tbody>
</table>
<h2 id="slots">Slots</h2>
<table>
<thead>
<tr>
<th>Slot</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../accession_id/">accession_id</a></td>
<td>A provider assigned accession identifier for a given entity</td>
</tr>
<tr>
<td><a href="../adj/">adj</a></td>
<td>Proportion of composition found in Adjacent Regions</td>
</tr>
<tr>
<td><a href="../annotations/">annotations</a></td>
<td></td>
</tr>
<tr>
<td><a href="../attribute_name/">attribute_name</a></td>
<td>The human-readable 'attribute name' can be set to a string which reflects its...</td>
</tr>
<tr>
<td><a href="../authority/">authority</a></td>
<td>The organization responsible for publishing the data</td>
</tr>
<tr>
<td><a href="../category/">category</a></td>
<td>Name of the high level ontology class in which this entity is categorized</td>
</tr>
<tr>
<td><a href="../cell_type_set_type/">cell_type_set_type</a></td>
<td>A tag denoting the type or purpose of the cell type set</td>
</tr>
<tr>
<td><a href="../checksum_algorithm/">checksum_algorithm</a></td>
<td>The type of cryptographic hash function used to calculate the checksum value</td>
</tr>
<tr>
<td><a href="../chembl_chirality/">chembl_chirality</a></td>
<td>Tern indicating the chirality of the chemical entity</td>
</tr>
<tr>
<td><a href="../chembl_drug_warning/">chembl_drug_warning</a></td>
<td>Text describing warnings for use of chemicals as therapeutics</td>
</tr>
<tr>
<td><a href="../chembl_prodrug/">chembl_prodrug</a></td>
<td>Flag indicating if a drug is a prodrug that is active only after being metabo...</td>
</tr>
<tr>
<td><a href="../color/">color</a></td>
<td>A string representing to hex triplet code of a color</td>
</tr>
<tr>
<td><a href="../color_hex_triplet/">color_hex_triplet</a></td>
<td>A hex string representing the display color for an associated entity</td>
</tr>
<tr>
<td><a href="../contains_cluster/">contains_cluster</a></td>
<td>One of potentially cell clusters which together defines the cell type taxon</td>
</tr>
<tr>
<td><a href="../contains_observation/">contains_observation</a></td>
<td>One of potentially many input observations which together define the cluster</td>
</tr>
<tr>
<td><a href="../contains_sample/">contains_sample</a></td>
<td>One of potentially many sample which together define the cluster</td>
</tr>
<tr>
<td><a href="../contains_taxon/">contains_taxon</a></td>
<td>One of potentially many cell types taxons which together defines the cell typ...</td>
</tr>
<tr>
<td><a href="../content_url/">content_url</a></td>
<td></td>
</tr>
<tr>
<td><a href="../creation_date/">creation_date</a></td>
<td>date on which an entity was created</td>
</tr>
<tr>
<td><a href="../curated_markers_to_mouse/">curated_markers_to_mouse</a></td>
<td>Marker genes in the mouse species that are related to the respective cell typ...</td>
</tr>
<tr>
<td><a href="../curated_markers_to_primates/">curated_markers_to_primates</a></td>
<td>Marker genes in the primate species that are related to the respective cell t...</td>
</tr>
<tr>
<td><a href="../denotes_cell_type/">denotes_cell_type</a></td>
<td>One of potentially many cell type terms  which the abbreviation denotes</td>
</tr>
<tr>
<td><a href="../denotes_gene_annotation/">denotes_gene_annotation</a></td>
<td>One of potentially many gene annotation terms to which the abbreviation denot...</td>
</tr>
<tr>
<td><a href="../denotes_parcellation_term/">denotes_parcellation_term</a></td>
<td>One of potentially many parcellation terms (anatomical structures) which the ...</td>
</tr>
<tr>
<td><a href="../deprecated/">deprecated</a></td>
<td>A boolean flag indicating that an entity is no longer considered current or v...</td>
</tr>
<tr>
<td><a href="../description/">description</a></td>
<td>a human-readable description of an entity</td>
</tr>
<tr>
<td><a href="../digest/">digest</a></td>
<td>Stores checksum information</td>
</tr>
<tr>
<td><a href="../entity_type/">entity_type</a></td>
<td>The entity type which the abbreviation term denotes</td>
</tr>
<tr>
<td><a href="../equivalent_identifiers/">equivalent_identifiers</a></td>
<td>A set of identifiers that are considered equivalent to the primary identifier...</td>
</tr>
<tr>
<td><a href="../format/">format</a></td>
<td></td>
</tr>
<tr>
<td><a href="../full_name/">full_name</a></td>
<td>a long-form human readable name for a thing</td>
</tr>
<tr>
<td><a href="../genome_annotations/">genome_annotations</a></td>
<td></td>
</tr>
<tr>
<td><a href="../genome_assemblies/">genome_assemblies</a></td>
<td></td>
</tr>
<tr>
<td><a href="../gpe/">gpe</a></td>
<td>Proportion of composition found in External Globus Pallidus</td>
</tr>
<tr>
<td><a href="../gpi/">gpi</a></td>
<td>Proportion of composition found in Internal Globus Pallidus</td>
</tr>
<tr>
<td><a href="../has_attribute/">has_attribute</a></td>
<td>connects any entity to an attribute</td>
</tr>
<tr>
<td><a href="../has_attribute_type/">has_attribute_type</a></td>
<td>connects an attribute to a class that describes it</td>
</tr>
<tr>
<td><a href="../has_biological_sequence/">has_biological_sequence</a></td>
<td>connects a genomic feature to its sequence</td>
</tr>
<tr>
<td><a href="../has_numeric_value/">has_numeric_value</a></td>
<td>connects a quantity value to a number</td>
</tr>
<tr>
<td><a href="../has_qualitative_value/">has_qualitative_value</a></td>
<td>connects an attribute to a value</td>
</tr>
<tr>
<td><a href="../has_quantitative_value/">has_quantitative_value</a></td>
<td>connects an attribute to a value</td>
</tr>
<tr>
<td><a href="../has_taxonomic_rank/">has_taxonomic_rank</a></td>
<td></td>
</tr>
<tr>
<td><a href="../has_unit/">has_unit</a></td>
<td>connects a quantity value to a unit</td>
</tr>
<tr>
<td><a href="../has_abbreviation/">has_abbreviation</a></td>
<td></td>
</tr>
<tr>
<td><a href="../has_anatomical_annotation_set/">has_anatomical_annotation_set</a></td>
<td>Reference to the anatomical annotation set component of the parcellation atla...</td>
</tr>
<tr>
<td><a href="../has_anatomical_space/">has_anatomical_space</a></td>
<td>Reference to the anatomical space component of the parcellation atlas</td>
</tr>
<tr>
<td><a href="../has_parcellation_terminology/">has_parcellation_terminology</a></td>
<td>Reference to the parcellation terminology component of the parcellation atlas</td>
</tr>
<tr>
<td><a href="../has_parent/">has_parent</a></td>
<td></td>
</tr>
<tr>
<td><a href="../has_parent_parcellation_term/">has_parent_parcellation_term</a></td>
<td>Reference to the parent parcellation term for which the parcellation term is ...</td>
</tr>
<tr>
<td><a href="../has_parent_parcellation_term_set/">has_parent_parcellation_term_set</a></td>
<td>Reference to the parent parcellation term set for which the parcellation term...</td>
</tr>
<tr>
<td><a href="../has_variable/">has_variable</a></td>
<td>One of set of genes which together forms the variable set of an observation m...</td>
</tr>
<tr>
<td><a href="../id/">id</a></td>
<td>A unique identifier for an entity</td>
</tr>
<tr>
<td><a href="../in_taxon/">in_taxon</a></td>
<td>connects an entity to its taxonomic classification</td>
</tr>
<tr>
<td><a href="../in_taxon_label/">in_taxon_label</a></td>
<td>The human readable scientific name for the taxon of the entity</td>
</tr>
<tr>
<td><a href="../information_content/">information_content</a></td>
<td>Information content (IC) value for a term, primarily from Automats</td>
</tr>
<tr>
<td><a href="../internal_identifier/">internal_identifier</a></td>
<td>An identifier that uniquely denotes a specific parcellation annotation within...</td>
</tr>
<tr>
<td><a href="../iri/">iri</a></td>
<td>An IRI for an entity</td>
</tr>
<tr>
<td><a href="../is_color_for_set/">is_color_for_set</a></td>
<td>The associated cell type set of the specified display color</td>
</tr>
<tr>
<td><a href="../is_color_for_taxon/">is_color_for_taxon</a></td>
<td>The associated cell type taxon for the specified display color</td>
</tr>
<tr>
<td><a href="../is_palette_for/">is_palette_for</a></td>
<td>The cell types taxonomy for which the palette provides display color properti...</td>
</tr>
<tr>
<td><a href="../is_revision_of/">is_revision_of</a></td>
<td></td>
</tr>
<tr>
<td><a href="../label/">label</a></td>
<td></td>
</tr>
<tr>
<td><a href="../license/">license</a></td>
<td></td>
</tr>
<tr>
<td><a href="../macromolecular_machine_mixin_name/">macromolecular_machine_mixin_name</a></td>
<td>genes are typically designated by a short symbol and a full name</td>
</tr>
<tr>
<td><a href="../meaning/">meaning</a></td>
<td>The meaning of an abbreviation term</td>
</tr>
<tr>
<td><a href="../measures/">measures</a></td>
<td>Reference to the specific image dataset used to define the anatomical space</td>
</tr>
<tr>
<td><a href="../molecular_type/">molecular_type</a></td>
<td></td>
</tr>
<tr>
<td><a href="../name/">name</a></td>
<td>A human-readable name for an attribute or entity</td>
</tr>
<tr>
<td><a href="../named_thing_category/">named_thing_category</a></td>
<td>Name of the high level ontology class in which this entity is categorized</td>
</tr>
<tr>
<td><a href="../node_property/">node_property</a></td>
<td>A grouping for any property that holds between a node and a value</td>
</tr>
<tr>
<td><a href="../number_of_cells/">number_of_cells</a></td>
<td>The aggregated number of cells that defines the cell type taxon</td>
</tr>
<tr>
<td><a href="../number_of_observations/">number_of_observations</a></td>
<td>The number of observations in the cluster</td>
</tr>
<tr>
<td><a href="../order/">order</a></td>
<td>The priority or display order of an entity among all other entiites of the sa...</td>
</tr>
<tr>
<td><a href="../ordinal/">ordinal</a></td>
<td></td>
</tr>
<tr>
<td><a href="../organism_taxon_has_taxonomic_rank/">organism_taxon_has_taxonomic_rank</a></td>
<td></td>
</tr>
<tr>
<td><a href="../parameterizes/">parameterizes</a></td>
<td>Reference to the anatomical space for which the anatomical annotation set is ...</td>
</tr>
<tr>
<td><a href="../part_of_anatomical_annotation_set/">part_of_anatomical_annotation_set</a></td>
<td></td>
</tr>
<tr>
<td><a href="../part_of_matrix/">part_of_matrix</a></td>
<td>The observation matrix for which is observation row is part of</td>
</tr>
<tr>
<td><a href="../part_of_palette/">part_of_palette</a></td>
<td>The color palette for which the display color map is part of</td>
</tr>
<tr>
<td><a href="../part_of_parcellation_color_scheme/">part_of_parcellation_color_scheme</a></td>
<td>Reference to the parcellation color scheme for which the color assignment is ...</td>
</tr>
<tr>
<td><a href="../part_of_parcellation_term_set/">part_of_parcellation_term_set</a></td>
<td>Reference to the parcellation term set for which the parcellation term is par...</td>
</tr>
<tr>
<td><a href="../part_of_parcellation_terminology/">part_of_parcellation_terminology</a></td>
<td>Reference to the parcellation terminology for which the parcellation term set...</td>
</tr>
<tr>
<td><a href="../part_of_set/">part_of_set</a></td>
<td>The cluster set from which the cluster is part of</td>
</tr>
<tr>
<td><a href="../part_of_taxonomy/">part_of_taxonomy</a></td>
<td></td>
</tr>
<tr>
<td><a href="../provided_by/">provided_by</a></td>
<td>The value in this node property represents the knowledge provider that create...</td>
</tr>
<tr>
<td><a href="../reference_assembly/">reference_assembly</a></td>
<td>The reference genome assembly that this genome annotation was created from</td>
</tr>
<tr>
<td><a href="../referenced_in/">referenced_in</a></td>
<td>The genome annotation that this gene was referenced from</td>
</tr>
<tr>
<td><a href="../related_to/">related_to</a></td>
<td>A relationship that is asserted between two named things</td>
</tr>
<tr>
<td><a href="../related_to_at_instance_level/">related_to_at_instance_level</a></td>
<td>Represents a relationship held between two instances of a data classes</td>
</tr>
<tr>
<td><a href="../represented_by/">represented_by</a></td>
<td>One of potentially matrix files which together represents the entire observat...</td>
</tr>
<tr>
<td><a href="../represented_in/">represented_in</a></td>
<td>The specific file where this observation row is represented</td>
</tr>
<tr>
<td><a href="../revision_of/">revision_of</a></td>
<td></td>
</tr>
<tr>
<td><a href="../rights/">rights</a></td>
<td></td>
</tr>
<tr>
<td><a href="../sn/">sn</a></td>
<td>Proportion of composition found in Substantia Nigra</td>
</tr>
<tr>
<td><a href="../source_id/">source_id</a></td>
<td>The authority specific identifier</td>
</tr>
<tr>
<td><a href="../spatial_proportions_human/">spatial_proportions_human</a></td>
<td>Quantification of regional composition (how much of something is found in eac...</td>
</tr>
<tr>
<td><a href="../spatial_proportions_macaque/">spatial_proportions_macaque</a></td>
<td>Quantification of regional composition (how much of something is found in eac...</td>
</tr>
<tr>
<td><a href="../spatial_proportions_marmoset/">spatial_proportions_marmoset</a></td>
<td>Quantification of regional composition (how much of something is found in eac...</td>
</tr>
<tr>
<td><a href="../spatial_regional_proportions/">spatial_regional_proportions</a></td>
<td>Species-independent quantification of regional composition (how much of somet...</td>
</tr>
<tr>
<td><a href="../specialization_of/">specialization_of</a></td>
<td>Reference to the general (non versioned) parcellation atlas for which the par...</td>
</tr>
<tr>
<td><a href="../sth/">sth</a></td>
<td>Proportion of composition found in Subthalamic Nucleus</td>
</tr>
<tr>
<td><a href="../str/">str</a></td>
<td>Proportion of composition found in Striatum</td>
</tr>
<tr>
<td><a href="../strain/">strain</a></td>
<td>The genetic variant or subtype of a species or organism</td>
</tr>
<tr>
<td><a href="../subject_parcellation_annotation/">subject_parcellation_annotation</a></td>
<td>Reference to the parcellation annotation that is the subject of the associati...</td>
</tr>
<tr>
<td><a href="../subject_parcellation_term/">subject_parcellation_term</a></td>
<td>Reference to the parcellation term that is the subject of the association</td>
</tr>
<tr>
<td><a href="../subject_parcellation_terminology/">subject_parcellation_terminology</a></td>
<td>Reference to the parcellation terminology for which the parcellation color sc...</td>
</tr>
<tr>
<td><a href="../symbol/">symbol</a></td>
<td>Symbol for a particular thing</td>
</tr>
<tr>
<td><a href="../synonym/">synonym</a></td>
<td>Alternate human-readable names for a thing</td>
</tr>
<tr>
<td><a href="../term/">term</a></td>
<td>An abbreviation term</td>
</tr>
<tr>
<td><a href="../type/">type</a></td>
<td></td>
</tr>
<tr>
<td><a href="../unit/">unit</a></td>
<td>A controlled vocabulary attribute defining the length unit of the x, y, and z...</td>
</tr>
<tr>
<td><a href="../used/">used</a></td>
<td>Usage is the beginning of utilizing an entity by an activity</td>
</tr>
<tr>
<td><a href="../value/">value</a></td>
<td>The checksum value obtained from a specific cryotographic hash function</td>
</tr>
<tr>
<td><a href="../version/">version</a></td>
<td></td>
</tr>
<tr>
<td><a href="../voxel_count/">voxel_count</a></td>
<td>The number of voxels (3D pixels) spanned by the parcellation annotation (opti...</td>
</tr>
<tr>
<td><a href="../was_derived_from/">was_derived_from</a></td>
<td>A derivation is a transformation of an entity into another, an update of an e...</td>
</tr>
<tr>
<td><a href="../was_generated_by/">was_generated_by</a></td>
<td>Generation is the completion of production of a new entity by an activity</td>
</tr>
<tr>
<td><a href="../x_direction/">x_direction</a></td>
<td>A controlled vocabulary attribute defining the x axis direction in terms of a...</td>
</tr>
<tr>
<td><a href="../x_resolution/">x_resolution</a></td>
<td>The resolution (length / pixel) in along the x axis (numerical value part)</td>
</tr>
<tr>
<td><a href="../x_size/">x_size</a></td>
<td>The number of pixels/voxels (size) along the x axis</td>
</tr>
<tr>
<td><a href="../xref/">xref</a></td>
<td>A database cross reference or alternative identifier for a NamedThing or edge...</td>
</tr>
<tr>
<td><a href="../y_direction/">y_direction</a></td>
<td>A controlled vocabulary attribute defining the y axis direction in terms of a...</td>
</tr>
<tr>
<td><a href="../y_resolution/">y_resolution</a></td>
<td>The resolution (length / pixel) in along the y axis (numerical value part)</td>
</tr>
<tr>
<td><a href="../y_size/">y_size</a></td>
<td>The number of pixels/voxels (size) along the y axis</td>
</tr>
<tr>
<td><a href="../z_direction/">z_direction</a></td>
<td>A controlled vocabulary attribute defining the z axis direction in terms of a...</td>
</tr>
<tr>
<td><a href="../z_resolution/">z_resolution</a></td>
<td>The resolution (length / pixel) in along the z axis (numerical value part)</td>
</tr>
<tr>
<td><a href="../z_size/">z_size</a></td>
<td>The number of pixels/voxels (size) along the y axis</td>
</tr>
</tbody>
</table>
<h2 id="enumerations">Enumerations</h2>
<table>
<thead>
<tr>
<th>Enumeration</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../AbbreviationEntityType/">AbbreviationEntityType</a></td>
<td></td>
</tr>
<tr>
<td><a href="../ANATOMICALDIRECTION/">ANATOMICALDIRECTION</a></td>
<td>A controlled vocabulary term defining axis direction in terms of anatomical d...</td>
</tr>
<tr>
<td><a href="../AuthorityType/">AuthorityType</a></td>
<td></td>
</tr>
<tr>
<td><a href="../BioType/">BioType</a></td>
<td></td>
</tr>
<tr>
<td><a href="../CellTypeSetType/">CellTypeSetType</a></td>
<td></td>
</tr>
<tr>
<td><a href="../DigestType/">DigestType</a></td>
<td></td>
</tr>
<tr>
<td><a href="../DISTANCEUNIT/">DISTANCEUNIT</a></td>
<td></td>
</tr>
<tr>
<td><a href="../Region/">Region</a></td>
<td></td>
</tr>
</tbody>
</table>
<h2 id="types">Types</h2>
<table>
<thead>
<tr>
<th>Type</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../BiologicalSequence/">BiologicalSequence</a></td>
<td></td>
</tr>
<tr>
<td><a href="../Boolean/">Boolean</a></td>
<td>A binary (true or false) value</td>
</tr>
<tr>
<td><a href="../Curie/">Curie</a></td>
<td>a compact URI</td>
</tr>
<tr>
<td><a href="../Date/">Date</a></td>
<td>a date (year, month and day) in an idealized calendar</td>
</tr>
<tr>
<td><a href="../DateOrDatetime/">DateOrDatetime</a></td>
<td>Either a date or a datetime</td>
</tr>
<tr>
<td><a href="../Datetime/">Datetime</a></td>
<td>The combination of a date and time</td>
</tr>
<tr>
<td><a href="../Decimal/">Decimal</a></td>
<td>A real number with arbitrary precision that conforms to the xsd:decimal speci...</td>
</tr>
<tr>
<td><a href="../Double/">Double</a></td>
<td>A real number that conforms to the xsd:double specification</td>
</tr>
<tr>
<td><a href="../Float/">Float</a></td>
<td>A real number that conforms to the xsd:float specification</td>
</tr>
<tr>
<td><a href="../Integer/">Integer</a></td>
<td>An integer</td>
</tr>
<tr>
<td><a href="../IriType/">IriType</a></td>
<td>An IRI</td>
</tr>
<tr>
<td><a href="../Jsonpath/">Jsonpath</a></td>
<td>A string encoding a JSON Path</td>
</tr>
<tr>
<td><a href="../Jsonpointer/">Jsonpointer</a></td>
<td>A string encoding a JSON Pointer</td>
</tr>
<tr>
<td><a href="../LabelType/">LabelType</a></td>
<td>A string that provides a human-readable name for an entity</td>
</tr>
<tr>
<td><a href="../NarrativeText/">NarrativeText</a></td>
<td>A string that provides a human-readable description of something</td>
</tr>
<tr>
<td><a href="../Ncname/">Ncname</a></td>
<td>Prefix part of CURIE</td>
</tr>
<tr>
<td><a href="../Nodeidentifier/">Nodeidentifier</a></td>
<td>A URI, CURIE or BNODE that represents a node in a model</td>
</tr>
<tr>
<td><a href="../Objectidentifier/">Objectidentifier</a></td>
<td>A URI or CURIE that represents an object in the model</td>
</tr>
<tr>
<td><a href="../Sparqlpath/">Sparqlpath</a></td>
<td>A string encoding a SPARQL Property Path</td>
</tr>
<tr>
<td><a href="../String/">String</a></td>
<td>A character string</td>
</tr>
<tr>
<td><a href="../SymbolType/">SymbolType</a></td>
<td></td>
</tr>
<tr>
<td><a href="../Time/">Time</a></td>
<td>A time object represents a (local) time of day, independent of any particular...</td>
</tr>
<tr>
<td><a href="../Unit/">Unit</a></td>
<td></td>
</tr>
<tr>
<td><a href="../Uri/">Uri</a></td>
<td>a complete URI</td>
</tr>
<tr>
<td><a href="../Uriorcurie/">Uriorcurie</a></td>
<td>a URI or a CURIE</td>
</tr>
</tbody>
</table>
<h2 id="subsets">Subsets</h2>
<table>
<thead>
<tr>
<th>Subset</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td><a href="../ModelOrganismDatabase/">ModelOrganismDatabase</a></td>
<td>Subset that is relevant for a typical Model Organism Database (MOD)</td>
</tr>
<tr>
<td><a href="../Samples/">Samples</a></td>
<td>Sample/biosample datamodel</td>
</tr>
<tr>
<td><a href="../Testing/">Testing</a></td>
<td>TBD</td>
</tr>
<tr>
<td><a href="../TranslatorMinimal/">TranslatorMinimal</a></td>
<td>Minimum subset of translator work</td>
</tr>
</tbody>
</table>












                
              </article>
            </div>
          
          
  <script>var tabs=__md_get("__tabs");if(Array.isArray(tabs))e:for(var set of document.querySelectorAll(".tabbed-set")){var labels=set.querySelector(".tabbed-labels");for(var tab of tabs)for(var label of labels.getElementsByTagName("label"))if(label.innerText.trim()===tab){var input=document.getElementById(label.htmlFor);input.checked=!0;continue e}}</script>

<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    
      
      
      <script id="__config" type="application/json">{"annotate": null, "base": "..", "features": ["content.tabs.link"], "search": "../assets/javascripts/workers/search.2c215733.min.js", "tags": null, "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}, "version": null}</script>
    
    
      <script src="../assets/javascripts/bundle.79ae519e.min.js"></script>
      
    
  </body>
</html>